// civ7-modding-tools@1.3.0/src/constants/COLLECTION.ts
export const COLLECTION = {
    ALL_CAPITAL_CITIES: 'COLLECTION_ALL_CAPITAL_CITIES',
    ALL_CAPTURED_CITIES: 'COLLECTION_ALL_CAPTURED_CITIES',
    ALL_CITIES: 'COLLECTION_ALL_CITIES',
    ALL_DISTRICTS: 'COLLECTION_ALL_DISTRICTS',
    ALL_PLAYERS: 'COLLECTION_ALL_PLAYERS',
    ALL_PLOT_YIELDS: 'COLLECTION_ALL_PLOT_YIELDS',
    ALL_TEAMS: 'COLLECTION_ALL_TEAMS',
    ALL_UNITS: 'COLLECTION_ALL_UNITS',
    ANY_CITY_AT_STORY: 'COLLECTION_ANY_CITY_AT_STORY',
    CITIES_FOLLOWING_OWNER_RELIGION: 'COLLECTION_CITIES_FOLLOWING_OWNER_RELIGION',
    CITY_CONSTRUCTIBLES: 'COLLECTION_CITY_CONSTRUCTIBLES',
    CITY_DISTRICTS: 'COLLECTION_CITY_DISTRICTS',
    CITY_PLOT_YIELDS: 'COLLECTION_CITY_PLOT_YIELDS',
    CITY_TRAINED_UNITS: 'COLLECTION_CITY_TRAINED_UNITS',
    COMMANDER_PACKED_UNITS: 'COLLECTION_COMMANDER_PACKED_UNITS',
    INDEPENDENT_NEAREST_STORY: 'COLLECTION_INDEPENDENT_NEAREST_STORY',
    INDEPENDENT_PLAYERS: 'COLLECTION_INDEPENDENT_PLAYERS',
    MAJOR_OPPOSING_PLAYERS: 'COLLECTION_MAJOR_OPPOSING_PLAYERS',
    MAJOR_OPPOSING_TEAMS: 'COLLECTION_MAJOR_OPPOSING_TEAMS',
    MAJOR_PLAYERS: 'COLLECTION_MAJOR_PLAYERS',
    MAJOR_TEAMS: 'COLLECTION_MAJOR_TEAMS',
    MINOR_PLAYERS: 'COLLECTION_MINOR_PLAYERS',
    NARRATIVE_STORY: 'COLLECTION_NARRATIVE_STORY',
    OPPONENT_AT_STORY: 'COLLECTION_OPPONENT_AT_STORY',
    OWNER: 'COLLECTION_OWNER',
    OWNER_ARMY: 'COLLECTION_OWNER_ARMY',
    OWNER_CITY: 'COLLECTION_OWNER_CITY',
    OWNER_CITY_NEAREST_STORY: 'COLLECTION_OWNER_CITY_NEAREST_STORY',
    OWNER_COMMANDER_HIGHEST_LEVEL: 'COLLECTION_OWNER_COMMANDER_HIGHEST_LEVEL',
    OWNER_COMMANDER_NEAREST_STORY: 'COLLECTION_OWNER_COMMANDER_NEAREST_STORY',
    OWNER_DISTANT_LANDS_CITY_MOST_TREASURE: 'COLLECTION_OWNER_DISTANT_LANDS_CITY_MOST_TREASURE',
    OWNER_DISTANT_LANDS_CITY_NEAREST_CAPITAL: 'COLLECTION_OWNER_DISTANT_LANDS_CITY_NEAREST_CAPITAL',
    OWNER_PLAYER: 'COLLECTION_OWNER_PLAYER',
    OWNER_UNIT: 'COLLECTION_OWNER_UNIT',
    OWNER_UNIT_NEAREST_STORY: 'COLLECTION_OWNER_UNIT_NEAREST_STORY',
    PLAYER_ARMIES: 'COLLECTION_PLAYER_ARMIES',
    PLAYER_BUILT_CITIES: 'COLLECTION_PLAYER_BUILT_CITIES',
    PLAYER_CAPITAL_CITY: 'COLLECTION_PLAYER_CAPITAL_CITY',
    PLAYER_CAPITAL_CITY_DISTRICTS: 'COLLECTION_PLAYER_CAPITAL_CITY_DISTRICTS',
    PLAYER_CAPTURED_CITIES: 'COLLECTION_PLAYER_CAPTURED_CITIES',
    PLAYER_CITIES: 'COLLECTION_PLAYER_CITIES',
    PLAYER_COMBAT: 'COLLECTION_PLAYER_COMBAT',
    PLAYER_CONSTRUCTIBLES: 'COLLECTION_PLAYER_CONSTRUCTIBLES',
    PLAYER_DISTRICTS: 'COLLECTION_PLAYER_DISTRICTS',
    PLAYER_PLOT_YIELDS: 'COLLECTION_PLAYER_PLOT_YIELDS',
    PLAYER_TRAINED_UNITS: 'COLLECTION_PLAYER_TRAINED_UNITS',
    PLAYER_UNITS: 'COLLECTION_PLAYER_UNITS',
    SINGLE_PLOT_YIELDS: 'COLLECTION_SINGLE_PLOT_YIELDS',
    TEAM_CAPITAL_CITIES: 'COLLECTION_TEAM_CAPITAL_CITIES',
    TEAM_CAPTURED_CITIES: 'COLLECTION_TEAM_CAPTURED_CITIES',
    TEAM_CITIES: 'COLLECTION_TEAM_CITIES',
    TEAM_DISTRICTS: 'COLLECTION_TEAM_DISTRICTS',
    TEAM_PLAYERS: 'COLLECTION_TEAM_PLAYERS',
    TEAM_UNITS: 'COLLECTION_TEAM_UNITS',
    UNIT_COMBAT: 'COLLECTION_UNIT_COMBAT',
    UNIT_NEAREST_OWNER_CITY: 'COLLECTION_UNIT_NEAREST_OWNER_CITY',
    UNIT_OCCUPIED_CITY: 'COLLECTION_UNIT_OCCUPIED_CITY',
    UNIT_OCCUPIED_DISTRICT: 'COLLECTION_UNIT_OCCUPIED_DISTRICT',
} as const;
