// civ7-modding-tools@1.3.0/src/constants/FEATURE.ts
export const FEATURE = {
    SAGEBRUSH_STEPPE: 'FEATURE_SAGEBRUSH_STEPPE',
    OASIS: 'FEATURE_OASIS',
    DESERT_FLOODPLAIN_MINOR: 'FEATURE_DESERT_FLOODPLAIN_MINOR',
    DESERT_FLOODPLAIN_NAVIGABLE: 'FEATURE_DESERT_FLOODPLAIN_NAVIGABLE',
    FOREST: 'FEATURE_FOREST',
    MARSH: 'FEATURE_MARSH',
    GRASSLAND_FLOODPLAIN_MINOR: 'FEATURE_GRASSLAND_FLOODPLAIN_MINOR',
    GRASSLAND_FLOODPLAIN_NAVIGABLE: 'FEATURE_GRASSLAND_FLOODPLAIN_NAVIGABLE',
    REEF: 'FEATURE_REEF',
    COLD_REEF: 'FEATURE_COLD_REEF',
    SAVANNA_WOODLAND: 'FEATURE_SAVANNA_WOODLAND',
    WATERING_HOLE: 'FEATURE_WATERING_HOLE',
    PLAINS_FLOODPLAIN_MINOR: 'FEATURE_PLAINS_FLOODPLAIN_MINOR',
    PLAINS_FLOODPLAIN_NAVIGABLE: 'FEATURE_PLAINS_FLOODPLAIN_NAVIGABLE',
    RAINFOREST: 'FEATURE_RAINFOREST',
    MANGROVE: 'FEATURE_MANGROVE',
    TROPICAL_FLOODPLAIN_MINOR: 'FEATURE_TROPICAL_FLOODPLAIN_MINOR',
    TROPICAL_FLOODPLAIN_NAVIGABLE: 'FEATURE_TROPICAL_FLOODPLAIN_NAVIGABLE',
    TAIGA: 'FEATURE_TAIGA',
    TUNDRA_BOG: 'FEATURE_TUNDRA_BOG',
    TUNDRA_FLOODPLAIN_MINOR: 'FEATURE_TUNDRA_FLOODPLAIN_MINOR',
    TUNDRA_FLOODPLAIN_NAVIGABLE: 'FEATURE_TUNDRA_FLOODPLAIN_NAVIGABLE',
    ICE: 'FEATURE_ICE',
    VOLCANO: 'FEATURE_VOLCANO',
    VALLEY_OF_FLOWERS: 'FEATURE_VALLEY_OF_FLOWERS',
    BARRIER_REEF: 'FEATURE_BARRIER_REEF',
    REDWOOD_FOREST: 'FEATURE_REDWOOD_FOREST',
    GRAND_CANYON: 'FEATURE_GRAND_CANYON',
    GULLFOSS: 'FEATURE_GULLFOSS',
    HOERIKWAGGO: 'FEATURE_HOERIKWAGGO',
    IGUAZU_FALLS: 'FEATURE_IGUAZU_FALLS',
    KILIMANJARO: 'FEATURE_KILIMANJARO',
    ZHANGJIAJIE: 'FEATURE_ZHANGJIAJIE',
    THERA: 'FEATURE_THERA',
    TORRES_DEL_PAINE: 'FEATURE_TORRES_DEL_PAINE',
    ULURU: 'FEATURE_ULURU',
} as const;
