// civ7-modding-tools@1.3.0/src/constants/REQUIREMENT.ts
export const REQUIREMENT = {
    AGE_CRISIS_STAGE_IS_X: 'REQUIREMENT_AGE_CRISIS_STAGE_IS_X',
    AGE_PROGRESSION_TOTAL_IS_X: 'REQUIREMENT_AGE_PROGRESSION_TOTAL_IS_X',
    ALLY: 'REQUIREMENT_ALLY',
    ALWAYS_MET: 'REQUIREMENT_ALWAYS_MET',
    ATTACKER_PLOT_ADJACENT_TO_COAST: 'REQUIREMENT_ATTACKER_PLOT_ADJACENT_TO_COAST',
    ATTACKER_PLOT_BIOME_MATCHES: 'REQUIREMENT_ATTACKER_PLOT_BIOME_MATCHES',
    ATTACKER_PLOT_IMPROVEMENT_DEFENSE_STRENGTH: 'REQUIREMENT_ATTACKER_PLOT_IMPROVEMENT_DEFENSE_STRENGTH',
    ATTACKER_PLOT_IMPROVEMENT_TYPE_MATCHES: 'REQUIREMENT_ATTACKER_PLOT_IMPROVEMENT_TYPE_MATCHES',
    BUILDING_IS_ADJACENT_TO_X: 'REQUIREMENT_BUILDING_IS_ADJACENT_TO_X',
    CITY_BUILT_TAGGED_CONSTRUCTIBLE: 'REQUIREMENT_CITY_BUILT_TAGGED_CONSTRUCTIBLE',
    CITY_CONVERTED: 'REQUIREMENT_CITY_CONVERTED',
    CITY_FOLLOWS_RELIGION: 'REQUIREMENT_CITY_FOLLOWS_RELIGION',
    CITY_FOUNDED_BY_UNIT_TYPE: 'REQUIREMENT_CITY_FOUNDED_BY_UNIT_TYPE',
    CITY_HAPPINESS: 'REQUIREMENT_CITY_HAPPINESS',
    CITY_HAS_ANY_BUILDING: 'REQUIREMENT_CITY_HAS_ANY_BUILDING',
    CITY_HAS_ANY_WONDER: 'REQUIREMENT_CITY_HAS_ANY_WONDER',
    CITY_HAS_BUILD_QUEUE: 'REQUIREMENT_CITY_HAS_BUILD_QUEUE',
    CITY_HAS_BUILDING: 'REQUIREMENT_CITY_HAS_BUILDING',
    CITY_HAS_FEATURE: 'REQUIREMENT_CITY_HAS_FEATURE',
    CITY_HAS_GARRISON_UNIT: 'REQUIREMENT_CITY_HAS_GARRISON_UNIT',
    CITY_HAS_GREAT_WORK: 'REQUIREMENT_CITY_HAS_GREAT_WORK',
    CITY_HAS_LEAST_POPULATION: 'REQUIREMENT_CITY_HAS_LEAST_POPULATION',
    CITY_HAS_ONLY_BUILDINGS_FROM_THIS_AGE: 'REQUIREMENT_CITY_HAS_ONLY_BUILDINGS_FROM_THIS_AGE',
    CITY_HAS_PROJECT: 'REQUIREMENT_CITY_HAS_PROJECT',
    CITY_HAS_RESOURCE: 'REQUIREMENT_CITY_HAS_RESOURCE',
    CITY_HAS_ROUTE: 'REQUIREMENT_CITY_HAS_ROUTE',
    CITY_HAS_TERRAIN: 'REQUIREMENT_CITY_HAS_TERRAIN',
    CITY_HAS_UNIQUE_QUARTER: 'REQUIREMENT_CITY_HAS_UNIQUE_QUARTER',
    CITY_HAS_X_OPEN_RESOURCE_SLOTS: 'REQUIREMENT_CITY_HAS_X_OPEN_RESOURCE_SLOTS',
    CITY_HAS_X_RESOURCES_ASSIGNED: 'REQUIREMENT_CITY_HAS_X_RESOURCES_ASSIGNED',
    CITY_IS_CAPITAL: 'REQUIREMENT_CITY_IS_CAPITAL',
    CITY_IS_CITY: 'REQUIREMENT_CITY_IS_CITY',
    CITY_IS_DISTANT_LANDS: 'REQUIREMENT_CITY_IS_DISTANT_LANDS',
    CITY_IS_INFECTED: 'REQUIREMENT_CITY_IS_INFECTED',
    CITY_IS_ORIGINAL_CAPITAL: 'REQUIREMENT_CITY_IS_ORIGINAL_CAPITAL',
    CITY_IS_ORIGINAL_OWNER: 'REQUIREMENT_CITY_IS_ORIGINAL_OWNER',
    CITY_IS_OWNER_CAPITAL_CONTINENT: 'REQUIREMENT_CITY_IS_OWNER_CAPITAL_CONTINENT',
    CITY_IS_OWNER_CAPITAL_HEMISPHERE: 'REQUIREMENT_CITY_IS_OWNER_CAPITAL_HEMISPHERE',
    CITY_IS_PRODUCING_PROJECT: 'REQUIREMENT_CITY_IS_PRODUCING_PROJECT',
    CITY_IS_TOWN: 'REQUIREMENT_CITY_IS_TOWN',
    CITY_LIBERATED: 'REQUIREMENT_CITY_LIBERATED',
    CITY_LOCATION_MATCHES: 'REQUIREMENT_CITY_LOCATION_MATCHES',
    CITY_OCCUPIED: 'REQUIREMENT_CITY_OCCUPIED',
    CITY_OCCUPIED_FRIEND: 'REQUIREMENT_CITY_OCCUPIED_FRIEND',
    CITY_OWNER_IS_SAME_TEAM: 'REQUIREMENT_CITY_OWNER_IS_SAME_TEAM',
    CITY_POPULATION: 'REQUIREMENT_CITY_POPULATION',
    CITY_RAZED: 'REQUIREMENT_CITY_RAZED',
    CITY_TRANSFER_TYPE_MATCHES: 'REQUIREMENT_CITY_TRANSFER_TYPE_MATCHES',
    CIVILIZATION_LEVEL: 'REQUIREMENT_CIVILIZATION_LEVEL',
    COLLECTION_ALL_MET: 'REQUIREMENT_COLLECTION_ALL_MET',
    COLLECTION_ANY_MET: 'REQUIREMENT_COLLECTION_ANY_MET',
    COLLECTION_COUNT_ATLEAST: 'REQUIREMENT_COLLECTION_COUNT_ATLEAST',
    COLLECTION_COUNT_EQUALS: 'REQUIREMENT_COLLECTION_COUNT_EQUALS',
    COLLECTION_COUNT_GREATERTHAN: 'REQUIREMENT_COLLECTION_COUNT_GREATERTHAN',
    COMBAT_TYPE_MATCHES: 'REQUIREMENT_COMBAT_TYPE_MATCHES',
    COMMANDER_EARNED_PROMOTION: 'REQUIREMENT_COMMANDER_EARNED_PROMOTION',
    COMMANDER_HAS_MAXED_DISCIPLINE: 'REQUIREMENT_COMMANDER_HAS_MAXED_DISCIPLINE',
    COMMANDER_HAS_X_PROMOTIONS: 'REQUIREMENT_COMMANDER_HAS_X_PROMOTIONS',
    COMMANDER_IN_COMMAND_RADIUS: 'REQUIREMENT_COMMANDER_IN_COMMAND_RADIUS',
    DECLARED_FRIEND: 'REQUIREMENT_DECLARED_FRIEND',
    DEFENDER_IS_PLAYER_OWNED_DISTRICT: 'REQUIREMENT_DEFENDER_IS_PLAYER_OWNED_DISTRICT',
    DEFENSIVE_PACT: 'REQUIREMENT_DEFENSIVE_PACT',
    DELEGATION: 'REQUIREMENT_DELEGATION',
    DEMAND_RECEIVED: 'REQUIREMENT_DEMAND_RECEIVED',
    DENOUNCED: 'REQUIREMENT_DENOUNCED',
    DENOUNCED_FRIEND: 'REQUIREMENT_DENOUNCED_FRIEND',
    ENGAGED_TARGET_OF_TARGET_MATCHES: 'REQUIREMENT_ENGAGED_TARGET_OF_TARGET_MATCHES',
    FOUNDED_NO_RELIGION: 'REQUIREMENT_FOUNDED_NO_RELIGION',
    GAME_AGE_ENDED: 'REQUIREMENT_GAME_AGE_ENDED',
    GAME_AGE_TYPE_MATCH: 'REQUIREMENT_GAME_AGE_TYPE_MATCH',
    GAME_ERA_ATLEAST: 'REQUIREMENT_GAME_ERA_ATLEAST',
    GAME_IS_MULTIPLAYER: 'REQUIREMENT_GAME_IS_MULTIPLAYER',
    GAME_IS_STARTED: 'REQUIREMENT_GAME_IS_STARTED',
    GAME_TURN_ATLEAST: 'REQUIREMENT_GAME_TURN_ATLEAST',
    GAME_TURN_ENDED: 'REQUIREMENT_GAME_TURN_ENDED',
    GAME_TURN_IS_LESS_THAN_X: 'REQUIREMENT_GAME_TURN_IS_LESS_THAN_X',
    GAME_TURN_IS_X: 'REQUIREMENT_GAME_TURN_IS_X',
    GAME_TURN_IS_X_NARRATIVE: 'REQUIREMENT_GAME_TURN_IS_X_NARRATIVE',
    GAME_TURN_MAX_REACHED: 'REQUIREMENT_GAME_TURN_MAX_REACHED',
    GAME_VICTORY_ENABLED: 'REQUIREMENT_GAME_VICTORY_ENABLED',
    GREAT_PERSON_TYPE_MATCHES: 'REQUIREMENT_GREAT_PERSON_TYPE_MATCHES',
    HAD_FRIENDLY_MEETING: 'REQUIREMENT_HAD_FRIENDLY_MEETING',
    IS_AGE_COUNT: 'REQUIREMENT_IS_AGE_COUNT',
    LEGACY_PATH_COMPLETED: 'REQUIREMENT_LEGACY_PATH_COMPLETED',
    LEGACY_PATH_MILESTONE_COMPLETED: 'REQUIREMENT_LEGACY_PATH_MILESTONE_COMPLETED',
    MAP_HAS_FEATURE: 'REQUIREMENT_MAP_HAS_FEATURE',
    MET: 'REQUIREMENT_MET',
    METAPROGRESSION_COMPLETED_CHALLENGES: 'REQUIREMENT_METAPROGRESSION_COMPLETED_CHALLENGES',
    METAPROGRESSION_LEGEND_PATH_LEVEL: 'REQUIREMENT_METAPROGRESSION_LEGEND_PATH_LEVEL',
    NEAR_RELIGIOUS_CITY: 'REQUIREMENT_NEAR_RELIGIOUS_CITY',
    NEAR_ROUTE: 'REQUIREMENT_NEAR_ROUTE',
    NEAR_TRADE_ROUTE: 'REQUIREMENT_NEAR_TRADE_ROUTE',
    NEVER_MET: 'REQUIREMENT_NEVER_MET',
    NOT_MET: 'REQUIREMENT_NOT_MET',
    OPEN_BORDERS: 'REQUIREMENT_OPEN_BORDERS',
    OPPONENT_IS_BARBARIAN: 'REQUIREMENT_OPPONENT_IS_BARBARIAN',
    OPPONENT_IS_DISTANT_LANDS: 'REQUIREMENT_OPPONENT_IS_DISTANT_LANDS',
    OPPONENT_IS_DISTRICT: 'REQUIREMENT_OPPONENT_IS_DISTRICT',
    OPPONENT_IS_FLANKING: 'REQUIREMENT_OPPONENT_IS_FLANKING',
    OPPONENT_IS_FORTIFIED: 'REQUIREMENT_OPPONENT_IS_FORTIFIED',
    OPPONENT_IS_INDEPENDENT: 'REQUIREMENT_OPPONENT_IS_INDEPENDENT',
    OPPONENT_IS_MINOR_CIV: 'REQUIREMENT_OPPONENT_IS_MINOR_CIV',
    OPPONENT_IS_OTHER_IDEOLOGY: 'REQUIREMENT_OPPONENT_IS_OTHER_IDEOLOGY',
    OPPONENT_IS_OTHER_RELIGION: 'REQUIREMENT_OPPONENT_IS_OTHER_RELIGION',
    OPPONENT_IS_SLOWER: 'REQUIREMENT_OPPONENT_IS_SLOWER',
    OPPONENT_IS_STRONGER: 'REQUIREMENT_OPPONENT_IS_STRONGER',
    OPPONENT_IS_WOUNDED: 'REQUIREMENT_OPPONENT_IS_WOUNDED',
    OPPONENT_UNIT_DOMAIN_MATCHES: 'REQUIREMENT_OPPONENT_UNIT_DOMAIN_MATCHES',
    OPPONENT_UNIT_TAG_MATCHES: 'REQUIREMENT_OPPONENT_UNIT_TAG_MATCHES',
    OPPONENT_UNIT_TYPE_MATCHES: 'REQUIREMENT_OPPONENT_UNIT_TYPE_MATCHES',
    PLAYER_ACTIVE_WARS: 'REQUIREMENT_PLAYER_ACTIVE_WARS',
    PLAYER_AT_PEACE_X_TURNS_AGO: 'REQUIREMENT_PLAYER_AT_PEACE_X_TURNS_AGO',
    PLAYER_AT_WAR_AND_HAS_MET: 'REQUIREMENT_PLAYER_AT_WAR_AND_HAS_MET',
    PLAYER_ATTACKED_CONTINENT: 'REQUIREMENT_PLAYER_ATTACKED_CONTINENT',
    PLAYER_BEHIND_TECH: 'REQUIREMENT_PLAYER_BEHIND_TECH',
    PLAYER_BUILT_WONDER: 'REQUIREMENT_PLAYER_BUILT_WONDER',
    PLAYER_CAN_EVER_EARN_GREAT_PERSON_CLASS: 'REQUIREMENT_PLAYER_CAN_EVER_EARN_GREAT_PERSON_CLASS',
    PLAYER_CARD_EFFECTS_COMPLETE: 'REQUIREMENT_PLAYER_CARD_EFFECTS_COMPLETE',
    PLAYER_CITY_IS_CLOSEST_TO_CITY: 'REQUIREMENT_PLAYER_CITY_IS_CLOSEST_TO_CITY',
    PLAYER_CITY_STATE_PROTECTOR: 'REQUIREMENT_PLAYER_CITY_STATE_PROTECTOR',
    PLAYER_CIVILIZATION_TAG_MATCHES: 'REQUIREMENT_PLAYER_CIVILIZATION_TAG_MATCHES',
    PLAYER_CIVILIZATION_TYPE_MATCHES: 'REQUIREMENT_PLAYER_CIVILIZATION_TYPE_MATCHES',
    PLAYER_CONQUERED_RELIGIOUS_CITY: 'REQUIREMENT_PLAYER_CONQUERED_RELIGIOUS_CITY',
    PLAYER_CONQUERED_SETTLEMENT_WITH_BUILDING: 'REQUIREMENT_PLAYER_CONQUERED_SETTLEMENT_WITH_BUILDING',
    PLAYER_CONQUERED_WONDER: 'REQUIREMENT_PLAYER_CONQUERED_WONDER',
    PLAYER_CONQUERED_X_CITIES: 'REQUIREMENT_PLAYER_CONQUERED_X_CITIES',
    PLAYER_CONQUERED_X_CITY_STATES: 'REQUIREMENT_PLAYER_CONQUERED_X_CITY_STATES',
    PLAYER_CONQUERING_ALLYING_MINORS: 'REQUIREMENT_PLAYER_CONQUERING_ALLYING_MINORS',
    PLAYER_CONVERTS_OPPONENTS_TO_FOUNDED_RELIGION: 'REQUIREMENT_PLAYER_CONVERTS_OPPONENTS_TO_FOUNDED_RELIGION',
    PLAYER_DARK_AGE: 'REQUIREMENT_PLAYER_DARK_AGE',
    PLAYER_DEAL_ENACTED: 'REQUIREMENT_PLAYER_DEAL_ENACTED',
    PLAYER_DECLARED_WAR: 'REQUIREMENT_PLAYER_DECLARED_WAR',
    PLAYER_DEFAULT_DEFEAT: 'REQUIREMENT_PLAYER_DEFAULT_DEFEAT',
    PLAYER_DIPLOMACY_ENGAGED_WITH_X_CIVS: 'REQUIREMENT_PLAYER_DIPLOMACY_ENGAGED_WITH_X_CIVS',
    PLAYER_DIPLOMACY_HAS_X_ALLIANCES: 'REQUIREMENT_PLAYER_DIPLOMACY_HAS_X_ALLIANCES',
    PLAYER_DIPLOMACY_INDEPENDENT_UPDATE_TYPE_MATCHES: 'REQUIREMENT_PLAYER_DIPLOMACY_INDEPENDENT_UPDATE_TYPE_MATCHES',
    PLAYER_DIPLOMACY_IS_INVOLVED_WITH_X_ACTIONS: 'REQUIREMENT_PLAYER_DIPLOMACY_IS_INVOLVED_WITH_X_ACTIONS',
    PLAYER_DISCOVERED_NATURAL_WONDER: 'REQUIREMENT_PLAYER_DISCOVERED_NATURAL_WONDER',
    PLAYER_DOMINATION_VICTORY: 'REQUIREMENT_PLAYER_DOMINATION_VICTORY',
    PLAYER_ELIGIBLE_CS_BONUS: 'REQUIREMENT_PLAYER_ELIGIBLE_CS_BONUS',
    PLAYER_ENVIRONMENT: 'REQUIREMENT_PLAYER_ENVIRONMENT',
    PLAYER_ERA_AT_LEAST: 'REQUIREMENT_PLAYER_ERA_AT_LEAST',
    PLAYER_ESTABLISHED_SETTLEMENT: 'REQUIREMENT_PLAYER_ESTABLISHED_SETTLEMENT',
    PLAYER_EXPLORATION_LEAD: 'REQUIREMENT_PLAYER_EXPLORATION_LEAD',
    PLAYER_EXTRACTS_ARTIFACT: 'REQUIREMENT_PLAYER_EXTRACTS_ARTIFACT',
    PLAYER_FIRST_TIME_SETTLEMENT_OCCUPATION: 'REQUIREMENT_PLAYER_FIRST_TIME_SETTLEMENT_OCCUPATION',
    PLAYER_FOUNDED_OUR_RELIGION: 'REQUIREMENT_PLAYER_FOUNDED_OUR_RELIGION',
    PLAYER_FOUNDED_RELIGION: 'REQUIREMENT_PLAYER_FOUNDED_RELIGION',
    PLAYER_FOUNDED_RELIGION_WITH_BELIEF: 'REQUIREMENT_PLAYER_FOUNDED_RELIGION_WITH_BELIEF',
    PLAYER_FOUNDS_SETTLEMENT_WITH_X_BIOME_TILES: 'REQUIREMENT_PLAYER_FOUNDS_SETTLEMENT_WITH_X_BIOME_TILES',
    PLAYER_GOT_GREAT_PERSON: 'REQUIREMENT_PLAYER_GOT_GREAT_PERSON',
    PLAYER_HANDICAP_AT: 'REQUIREMENT_PLAYER_HANDICAP_AT',
    PLAYER_HANDICAP_AT_OR_ABOVE: 'REQUIREMENT_PLAYER_HANDICAP_AT_OR_ABOVE',
    PLAYER_HAS_ACHIEVED_X_RELATIONSHIPS: 'REQUIREMENT_PLAYER_HAS_ACHIEVED_X_RELATIONSHIPS',
    PLAYER_HAS_ACTIVE_STORY: 'REQUIREMENT_PLAYER_HAS_ACTIVE_STORY',
    PLAYER_HAS_ACTIVE_TRADITION: 'REQUIREMENT_PLAYER_HAS_ACTIVE_TRADITION',
    PLAYER_HAS_ARCHIVED_GREAT_WORK: 'REQUIREMENT_PLAYER_HAS_ARCHIVED_GREAT_WORK',
    PLAYER_HAS_AT_LEAST_INDEPENDENT_RELATIONSHIP: 'REQUIREMENT_PLAYER_HAS_AT_LEAST_INDEPENDENT_RELATIONSHIP',
    PLAYER_HAS_AT_LEAST_NUM_BUILDINGS: 'REQUIREMENT_PLAYER_HAS_AT_LEAST_NUM_BUILDINGS',
    PLAYER_HAS_AT_LEAST_NUM_DIPLOMATIC_TOKENS: 'REQUIREMENT_PLAYER_HAS_AT_LEAST_NUM_DIPLOMATIC_TOKENS',
    PLAYER_HAS_AT_LEAST_NUM_GOSSIPS: 'REQUIREMENT_PLAYER_HAS_AT_LEAST_NUM_GOSSIPS',
    PLAYER_HAS_AT_LEAST_NUM_MILITARY_UNITS: 'REQUIREMENT_PLAYER_HAS_AT_LEAST_NUM_MILITARY_UNITS',
    PLAYER_HAS_AT_LEAST_NUM_NARRATIVE_TAGS: 'REQUIREMENT_PLAYER_HAS_AT_LEAST_NUM_NARRATIVE_TAGS',
    PLAYER_HAS_AT_LEAST_NUM_TRIBUTARIES: 'REQUIREMENT_PLAYER_HAS_AT_LEAST_NUM_TRIBUTARIES',
    PLAYER_HAS_AT_LEAST_NUM_TRIBUTARY_TYPES: 'REQUIREMENT_PLAYER_HAS_AT_LEAST_NUM_TRIBUTARY_TYPES',
    PLAYER_HAS_AT_LEAST_NUM_UNIT_TYPE: 'REQUIREMENT_PLAYER_HAS_AT_LEAST_NUM_UNIT_TYPE',
    PLAYER_HAS_AT_LEAST_NUM_UNLOCKS: 'REQUIREMENT_PLAYER_HAS_AT_LEAST_NUM_UNLOCKS',
    PLAYER_HAS_CITY_CENTER_X_DISTANCE_FROM_UNTOUCHED_NATURAL_WONDER: 'REQUIREMENT_PLAYER_HAS_CITY_CENTER_X_DISTANCE_FROM_UNTOUCHED_NATURAL_WONDER',
    PLAYER_HAS_CITY_REVOLT: 'REQUIREMENT_PLAYER_HAS_CITY_REVOLT',
    PLAYER_HAS_CITY_WITH_X_WORKERS: 'REQUIREMENT_PLAYER_HAS_CITY_WITH_X_WORKERS',
    PLAYER_HAS_CIVILIZATION_OR_LEADER_TRAIT: 'REQUIREMENT_PLAYER_HAS_CIVILIZATION_OR_LEADER_TRAIT',
    PLAYER_HAS_COMMANDERS_MAXED_DISCIPLINES: 'REQUIREMENT_PLAYER_HAS_COMMANDERS_MAXED_DISCIPLINES',
    PLAYER_HAS_COMMANDERS_WITH_X_LEVELS: 'REQUIREMENT_PLAYER_HAS_COMMANDERS_WITH_X_LEVELS',
    PLAYER_HAS_COMP_STOMP_VICTORY: 'REQUIREMENT_PLAYER_HAS_COMP_STOMP_VICTORY',
    PLAYER_HAS_COMPLETED_ATTRIBUTE_TREE: 'REQUIREMENT_PLAYER_HAS_COMPLETED_ATTRIBUTE_TREE',
    PLAYER_HAS_COMPLETED_NARRATIVE_STORY: 'REQUIREMENT_PLAYER_HAS_COMPLETED_NARRATIVE_STORY',
    PLAYER_HAS_COMPLETED_PROGRESSION_TREE_NODE: 'REQUIREMENT_PLAYER_HAS_COMPLETED_PROGRESSION_TREE_NODE',
    PLAYER_HAS_COMPLETED_PROJECT: 'REQUIREMENT_PLAYER_HAS_COMPLETED_PROJECT',
    PLAYER_HAS_COMPLETED_X_DIPLOMACY_ACTIONS: 'REQUIREMENT_PLAYER_HAS_COMPLETED_X_DIPLOMACY_ACTIONS',
    PLAYER_HAS_COMPLETED_X_PROGRESSION_TREE_NODES: 'REQUIREMENT_PLAYER_HAS_COMPLETED_X_PROGRESSION_TREE_NODES',
    PLAYER_HAS_COMPLETED_X_STORIES: 'REQUIREMENT_PLAYER_HAS_COMPLETED_X_STORIES',
    PLAYER_HAS_CONSTRUCTIBLE_FILLED_WITH_GREATWORKS: 'REQUIREMENT_PLAYER_HAS_CONSTRUCTIBLE_FILLED_WITH_GREATWORKS',
    PLAYER_HAS_CONVERTED_X_SETTLEMENTS: 'REQUIREMENT_PLAYER_HAS_CONVERTED_X_SETTLEMENTS',
    PLAYER_HAS_CURRENT_AGE_WONDER: 'REQUIREMENT_PLAYER_HAS_CURRENT_AGE_WONDER',
    PLAYER_HAS_ENGAGED_DIPLOMACY_ACTIONS: 'REQUIREMENT_PLAYER_HAS_ENGAGED_DIPLOMACY_ACTIONS',
    PLAYER_HAS_EXPLORED_X_TILES: 'REQUIREMENT_PLAYER_HAS_EXPLORED_X_TILES',
    PLAYER_HAS_FEATURE: 'REQUIREMENT_PLAYER_HAS_FEATURE',
    PLAYER_HAS_FEWEST_SETTLEMENTS: 'REQUIREMENT_PLAYER_HAS_FEWEST_SETTLEMENTS',
    PLAYER_HAS_FOUNDED_X_CITIES: 'REQUIREMENT_PLAYER_HAS_FOUNDED_X_CITIES',
    PLAYER_HAS_GAINED_YIELD: 'REQUIREMENT_PLAYER_HAS_GAINED_YIELD',
    PLAYER_HAS_GOVERNMENT: 'REQUIREMENT_PLAYER_HAS_GOVERNMENT',
    PLAYER_HAS_GREAT_PERSON: 'REQUIREMENT_PLAYER_HAS_GREAT_PERSON',
    PLAYER_HAS_GREAT_WORK_ACTIVE: 'REQUIREMENT_PLAYER_HAS_GREAT_WORK_ACTIVE',
    PLAYER_HAS_HIGHER_PROGRESSION: 'REQUIREMENT_PLAYER_HAS_HIGHER_PROGRESSION',
    PLAYER_HAS_HIGHEST_SCORE: 'REQUIREMENT_PLAYER_HAS_HIGHEST_SCORE',
    PLAYER_HAS_IDEOLOGY: 'REQUIREMENT_PLAYER_HAS_IDEOLOGY',
    PLAYER_HAS_IMPROVEMENT_HIT_WITH_FERTILITY: 'REQUIREMENT_PLAYER_HAS_IMPROVEMENT_HIT_WITH_FERTILITY',
    PLAYER_HAS_IMPROVEMENTS_ON_EVERY_NATURAL_WONDER: 'REQUIREMENT_PLAYER_HAS_IMPROVEMENTS_ON_EVERY_NATURAL_WONDER',
    PLAYER_HAS_MET: 'REQUIREMENT_PLAYER_HAS_MET',
    PLAYER_HAS_MOST_NATURAL_WONDER_IMPROVEMENTS: 'REQUIREMENT_PLAYER_HAS_MOST_NATURAL_WONDER_IMPROVEMENTS',
    PLAYER_HAS_MOST_RELIGION_FOLLOWERS: 'REQUIREMENT_PLAYER_HAS_MOST_RELIGION_FOLLOWERS',
    PLAYER_HAS_MOST_TRADE_ROUTES: 'REQUIREMENT_PLAYER_HAS_MOST_TRADE_ROUTES',
    PLAYER_HAS_MOST_WMD_CITY_HITS: 'REQUIREMENT_PLAYER_HAS_MOST_WMD_CITY_HITS',
    PLAYER_HAS_MOST_WONDERS: 'REQUIREMENT_PLAYER_HAS_MOST_WONDERS',
    PLAYER_HAS_NEGATIVE_YIELD_PER_TURN: 'REQUIREMENT_PLAYER_HAS_NEGATIVE_YIELD_PER_TURN',
    PLAYER_HAS_PANTHEON: 'REQUIREMENT_PLAYER_HAS_PANTHEON',
    PLAYER_HAS_PROGRESSION_TREE: 'REQUIREMENT_PLAYER_HAS_PROGRESSION_TREE',
    PLAYER_HAS_RESOURCE: 'REQUIREMENT_PLAYER_HAS_RESOURCE',
    PLAYER_HAS_RESOURCE_VISIBILITY: 'REQUIREMENT_PLAYER_HAS_RESOURCE_VISIBILITY',
    PLAYER_HAS_SAME_GOVERNMENT: 'REQUIREMENT_PLAYER_HAS_SAME_GOVERNMENT',
    PLAYER_HAS_SETTLEMENTS_WITH_POPULATION: 'REQUIREMENT_PLAYER_HAS_SETTLEMENTS_WITH_POPULATION',
    PLAYER_HAS_SPENT_AN_ATTRIBUTE_POINT: 'REQUIREMENT_PLAYER_HAS_SPENT_AN_ATTRIBUTE_POINT',
    PLAYER_HAS_SPENT_ATTRIBUTE_POINT: 'REQUIREMENT_PLAYER_HAS_SPENT_ATTRIBUTE_POINT',
    PLAYER_HAS_TRADE_ROUTES_FROM_X_PLAYERS: 'REQUIREMENT_PLAYER_HAS_TRADE_ROUTES_FROM_X_PLAYERS',
    PLAYER_HAS_UNHAPPY_CITIES: 'REQUIREMENT_PLAYER_HAS_UNHAPPY_CITIES',
    PLAYER_HAS_UNLOCK: 'REQUIREMENT_PLAYER_HAS_UNLOCK',
    PLAYER_HAS_VICTORY_POINTS_FOR_TYPE: 'REQUIREMENT_PLAYER_HAS_VICTORY_POINTS_FOR_TYPE',
    PLAYER_HAS_X_CITIES_WITH_BUILDING_SET: 'REQUIREMENT_PLAYER_HAS_X_CITIES_WITH_BUILDING_SET',
    PLAYER_HAS_X_DISTINCTIVE_BUILDINGS_FROM_CURRENT_AGE: 'REQUIREMENT_PLAYER_HAS_X_DISTINCTIVE_BUILDINGS_FROM_CURRENT_AGE',
    PLAYER_HAS_X_FILLED_CONSTRUCTIBLES: 'REQUIREMENT_PLAYER_HAS_X_FILLED_CONSTRUCTIBLES',
    PLAYER_HAS_X_GREAT_WORKS: 'REQUIREMENT_PLAYER_HAS_X_GREAT_WORKS',
    PLAYER_HAS_X_IDEOLOGICAL_CONQUESTS: 'REQUIREMENT_PLAYER_HAS_X_IDEOLOGICAL_CONQUESTS',
    PLAYER_HAS_X_IMPORTED_RESOURCES: 'REQUIREMENT_PLAYER_HAS_X_IMPORTED_RESOURCES',
    PLAYER_HAS_X_INFECTED_SETTLEMENTS: 'REQUIREMENT_PLAYER_HAS_X_INFECTED_SETTLEMENTS',
    PLAYER_HAS_X_MASTERIES: 'REQUIREMENT_PLAYER_HAS_X_MASTERIES',
    PLAYER_HAS_X_RELIGIONS: 'REQUIREMENT_PLAYER_HAS_X_RELIGIONS',
    PLAYER_HAS_X_RESOURCE_TYPE: 'REQUIREMENT_PLAYER_HAS_X_RESOURCE_TYPE',
    PLAYER_HAS_X_RESOURCES: 'REQUIREMENT_PLAYER_HAS_X_RESOURCES',
    PLAYER_HAS_X_RESOURCES_IN_ONE_CITY: 'REQUIREMENT_PLAYER_HAS_X_RESOURCES_IN_ONE_CITY',
    PLAYER_HAS_X_SETTLEMENTS: 'REQUIREMENT_PLAYER_HAS_X_SETTLEMENTS',
    PLAYER_HAS_X_SETTLEMENTS_AT: 'REQUIREMENT_PLAYER_HAS_X_SETTLEMENTS_AT',
    PLAYER_HAS_X_SETTLEMENTS_FOREIGN_HEMISPHERE: 'REQUIREMENT_PLAYER_HAS_X_SETTLEMENTS_FOREIGN_HEMISPHERE',
    PLAYER_HAS_X_SETTLEMENTS_IN_BIOME: 'REQUIREMENT_PLAYER_HAS_X_SETTLEMENTS_IN_BIOME',
    PLAYER_HAS_X_SETTLEMENTS_WITH_RELIGION: 'REQUIREMENT_PLAYER_HAS_X_SETTLEMENTS_WITH_RELIGION',
    PLAYER_HAS_X_STARTING_YIELD: 'REQUIREMENT_PLAYER_HAS_X_STARTING_YIELD',
    PLAYER_HAS_X_TOWNS_PRODUCING_PROJECT: 'REQUIREMENT_PLAYER_HAS_X_TOWNS_PRODUCING_PROJECT',
    PLAYER_HAS_X_TRADE_ROUTES_FROM_PLAYER: 'REQUIREMENT_PLAYER_HAS_X_TRADE_ROUTES_FROM_PLAYER',
    PLAYER_HAS_X_TRADE_ROUTES_WITH_PLAYER: 'REQUIREMENT_PLAYER_HAS_X_TRADE_ROUTES_WITH_PLAYER',
    PLAYER_HAS_X_UNIQUE_RESOURCES: 'REQUIREMENT_PLAYER_HAS_X_UNIQUE_RESOURCES',
    PLAYER_HAS_X_URBAN_DISTRICTS_WITH_YIELD: 'REQUIREMENT_PLAYER_HAS_X_URBAN_DISTRICTS_WITH_YIELD',
    PLAYER_HAS_X_VICTORY_POINTS: 'REQUIREMENT_PLAYER_HAS_X_VICTORY_POINTS',
    PLAYER_HAS_X_WAR_SUPPORT: 'REQUIREMENT_PLAYER_HAS_X_WAR_SUPPORT',
    PLAYER_HAS_X_WONDER_IMPROVEMENT_POINTS: 'REQUIREMENT_PLAYER_HAS_X_WONDER_IMPROVEMENT_POINTS',
    PLAYER_HAS_X_WONDERS: 'REQUIREMENT_PLAYER_HAS_X_WONDERS',
    PLAYER_HAS_X_WORKERS: 'REQUIREMENT_PLAYER_HAS_X_WORKERS',
    PLAYER_INITIATED_DIPLOMACY_ACTION_RESPONSE_TYPE_MATCHES: 'REQUIREMENT_PLAYER_INITIATED_DIPLOMACY_ACTION_RESPONSE_TYPE_MATCHES',
    PLAYER_IS_AT_PEACE: 'REQUIREMENT_PLAYER_IS_AT_PEACE',
    PLAYER_IS_AT_PEACE_WITH_ALL_MAJORS: 'REQUIREMENT_PLAYER_IS_AT_PEACE_WITH_ALL_MAJORS',
    PLAYER_IS_AT_WAR_WITH_OPPOSING_IDEOLOGY: 'REQUIREMENT_PLAYER_IS_AT_WAR_WITH_OPPOSING_IDEOLOGY',
    PLAYER_IS_ATTACKING: 'REQUIREMENT_PLAYER_IS_ATTACKING',
    PLAYER_IS_BARBARIAN: 'REQUIREMENT_PLAYER_IS_BARBARIAN',
    PLAYER_IS_CLOSE_TO_VICTORY: 'REQUIREMENT_PLAYER_IS_CLOSE_TO_VICTORY',
    PLAYER_IS_HUMAN: 'REQUIREMENT_PLAYER_IS_HUMAN',
    PLAYER_IS_IN_GOLDEN_AGE: 'REQUIREMENT_PLAYER_IS_IN_GOLDEN_AGE',
    PLAYER_IS_INITIATING_COMBAT: 'REQUIREMENT_PLAYER_IS_INITIATING_COMBAT',
    PLAYER_IS_MAJOR: 'REQUIREMENT_PLAYER_IS_MAJOR',
    PLAYER_IS_MINOR: 'REQUIREMENT_PLAYER_IS_MINOR',
    PLAYER_IS_RELIGION_FOUNDER: 'REQUIREMENT_PLAYER_IS_RELIGION_FOUNDER',
    PLAYER_IS_RESEARCHING_PROGRESSION_TREE_NODE: 'REQUIREMENT_PLAYER_IS_RESEARCHING_PROGRESSION_TREE_NODE',
    PLAYER_IS_SAME_RELIGION: 'REQUIREMENT_PLAYER_IS_SAME_RELIGION',
    PLAYER_IS_TEAM_MEMBER: 'REQUIREMENT_PLAYER_IS_TEAM_MEMBER',
    PLAYER_LEADER_TAG_MATCHES: 'REQUIREMENT_PLAYER_LEADER_TAG_MATCHES',
    PLAYER_LEADER_TYPE_MATCHES: 'REQUIREMENT_PLAYER_LEADER_TYPE_MATCHES',
    PLAYER_MAJORITY_RELIGION_IS_OWNER: 'REQUIREMENT_PLAYER_MAJORITY_RELIGION_IS_OWNER',
    PLAYER_MEETS_SCORE_THRESHOLD: 'REQUIREMENT_PLAYER_MEETS_SCORE_THRESHOLD',
    PLAYER_MET_NUM_CIVS: 'REQUIREMENT_PLAYER_MET_NUM_CIVS',
    PLAYER_MET_X_TURNS_AGO: 'REQUIREMENT_PLAYER_MET_X_TURNS_AGO',
    PLAYER_MILITARY_STRENGTH_LEAD: 'REQUIREMENT_PLAYER_MILITARY_STRENGTH_LEAD',
    PLAYER_NOT_INVOLVED_MINORS: 'REQUIREMENT_PLAYER_NOT_INVOLVED_MINORS',
    PLAYER_NUKE_LOVER: 'REQUIREMENT_PLAYER_NUKE_LOVER',
    PLAYER_ON_NON_HOME_CONTINENT: 'REQUIREMENT_PLAYER_ON_NON_HOME_CONTINENT',
    PLAYER_OVER_SETTLEMENT_CAP: 'REQUIREMENT_PLAYER_OVER_SETTLEMENT_CAP',
    PLAYER_OVERBUILDS: 'REQUIREMENT_PLAYER_OVERBUILDS',
    PLAYER_OWNS_OBJECT: 'REQUIREMENT_PLAYER_OWNS_OBJECT',
    PLAYER_PEACEFUL_ON_CONTINENT: 'REQUIREMENT_PLAYER_PEACEFUL_ON_CONTINENT',
    PLAYER_PLOTS_REVEALED_ARE_X: 'REQUIREMENT_PLAYER_PLOTS_REVEALED_ARE_X',
    PLAYER_POPULATION_LEAD: 'REQUIREMENT_PLAYER_POPULATION_LEAD',
    PLAYER_PRODUCES_AT_LEAST_X_RESOURCE: 'REQUIREMENT_PLAYER_PRODUCES_AT_LEAST_X_RESOURCE',
    PLAYER_RAZED_X_CITIES: 'REQUIREMENT_PLAYER_RAZED_X_CITIES',
    PLAYER_RECONQUERED_X_CITIES: 'REQUIREMENT_PLAYER_RECONQUERED_X_CITIES',
    PLAYER_RELIGION_RECEIVED: 'REQUIREMENT_PLAYER_RELIGION_RECEIVED',
    PLAYER_SEEN_STORYLET: 'REQUIREMENT_PLAYER_SEEN_STORYLET',
    PLAYER_SHARES_HOME_CONTINENT: 'REQUIREMENT_PLAYER_SHARES_HOME_CONTINENT',
    PLAYER_SPECIFIC_WONDER_BUILT: 'REQUIREMENT_PLAYER_SPECIFIC_WONDER_BUILT',
    PLAYER_STARTS_TRADE_ROUTE_WITH_MATCHING_BUILDING: 'REQUIREMENT_PLAYER_STARTS_TRADE_ROUTE_WITH_MATCHING_BUILDING',
    PLAYER_TOTAL_DISTRICTS: 'REQUIREMENT_PLAYER_TOTAL_DISTRICTS',
    PLAYER_TOTAL_IMPROVED_RESOURCES: 'REQUIREMENT_PLAYER_TOTAL_IMPROVED_RESOURCES',
    PLAYER_TOTAL_POPULATION: 'REQUIREMENT_PLAYER_TOTAL_POPULATION',
    PLAYER_TRADE_ROUTE_HAS_X_RESOURCES: 'REQUIREMENT_PLAYER_TRADE_ROUTE_HAS_X_RESOURCES',
    PLAYER_TRADES_FOR_X_RESOURCES: 'REQUIREMENT_PLAYER_TRADES_FOR_X_RESOURCES',
    PLAYER_TRADES_WITH_PLAYERS_AT_WAR: 'REQUIREMENT_PLAYER_TRADES_WITH_PLAYERS_AT_WAR',
    PLAYER_TRADES_WITH_X_PLAYERS: 'REQUIREMENT_PLAYER_TRADES_WITH_X_PLAYERS',
    PLAYER_TRADES_X_RESOURCES: 'REQUIREMENT_PLAYER_TRADES_X_RESOURCES',
    PLAYER_TRADITION_SLOTS: 'REQUIREMENT_PLAYER_TRADITION_SLOTS',
    PLAYER_TRIGGERED_DISCOVERY: 'REQUIREMENT_PLAYER_TRIGGERED_DISCOVERY',
    PLAYER_TURN_STARTED: 'REQUIREMENT_PLAYER_TURN_STARTED',
    PLAYER_TYPE_MATCHES: 'REQUIREMENT_PLAYER_TYPE_MATCHES',
    PLAYER_UNIT_DESTROYS_DISTRICT_DEFENSES: 'REQUIREMENT_PLAYER_UNIT_DESTROYS_DISTRICT_DEFENSES',
    PLAYER_UNLOCKED_CIVIC: 'REQUIREMENT_PLAYER_UNLOCKED_CIVIC',
    PLAYER_UNLOCKED_TECHNOLOGY: 'REQUIREMENT_PLAYER_UNLOCKED_TECHNOLOGY',
    PLAYER_VICTORY_COUNT_ACHIEVED: 'REQUIREMENT_PLAYER_VICTORY_COUNT_ACHIEVED',
    PLAYER_VICTORY_PLACED: 'REQUIREMENT_PLAYER_VICTORY_PLACED',
    PLAYER_VICTORY_PROGRESS: 'REQUIREMENT_PLAYER_VICTORY_PROGRESS',
    PLAYER_YIELD_LEAD: 'REQUIREMENT_PLAYER_YIELD_LEAD',
    PLOT_ADJACENT_CONSTRUCTIBLE_TYPE_MATCHES: 'REQUIREMENT_PLOT_ADJACENT_CONSTRUCTIBLE_TYPE_MATCHES',
    PLOT_ADJACENT_FEATURE_TYPE_MATCHES: 'REQUIREMENT_PLOT_ADJACENT_FEATURE_TYPE_MATCHES',
    PLOT_ADJACENT_FRIENDLY_UNIT_TAG_MATCHES: 'REQUIREMENT_PLOT_ADJACENT_FRIENDLY_UNIT_TAG_MATCHES',
    PLOT_ADJACENT_FRIENDLY_UNIT_TYPE_MATCHES: 'REQUIREMENT_PLOT_ADJACENT_FRIENDLY_UNIT_TYPE_MATCHES',
    PLOT_ADJACENT_IMPROVEMENT_TYPE_MATCHES: 'REQUIREMENT_PLOT_ADJACENT_IMPROVEMENT_TYPE_MATCHES',
    PLOT_ADJACENT_RESOURCE_CLASS_TYPE_MATCHES: 'REQUIREMENT_PLOT_ADJACENT_RESOURCE_CLASS_TYPE_MATCHES',
    PLOT_ADJACENT_TERRAIN_TYPE_MATCHES: 'REQUIREMENT_PLOT_ADJACENT_TERRAIN_TYPE_MATCHES',
    PLOT_ADJACENT_TO_CITY_WITH_X_RESOURCES_SLOTTED: 'REQUIREMENT_PLOT_ADJACENT_TO_CITY_WITH_X_RESOURCES_SLOTTED',
    PLOT_ADJACENT_TO_COAST: 'REQUIREMENT_PLOT_ADJACENT_TO_COAST',
    PLOT_ADJACENT_TO_CONSTRUCTIBLE_WITH_X_GREAT_WORKS: 'REQUIREMENT_PLOT_ADJACENT_TO_CONSTRUCTIBLE_WITH_X_GREAT_WORKS',
    PLOT_ADJACENT_TO_LAKE: 'REQUIREMENT_PLOT_ADJACENT_TO_LAKE',
    PLOT_ADJACENT_TO_OWNER: 'REQUIREMENT_PLOT_ADJACENT_TO_OWNER',
    PLOT_ADJACENT_TO_OWNER_AT_WAR: 'REQUIREMENT_PLOT_ADJACENT_TO_OWNER_AT_WAR',
    PLOT_ADJACENT_TO_RIVER: 'REQUIREMENT_PLOT_ADJACENT_TO_RIVER',
    PLOT_ADJACENT_TO_UNIQUE_QUARTER: 'REQUIREMENT_PLOT_ADJACENT_TO_UNIQUE_QUARTER',
    PLOT_ADJACENT_TO_WONDER: 'REQUIREMENT_PLOT_ADJACENT_TO_WONDER',
    PLOT_ADJACENT_UNIT_TAG_MATCHES: 'REQUIREMENT_PLOT_ADJACENT_UNIT_TAG_MATCHES',
    PLOT_ADJACENT_UNIT_TYPE_MATCHES: 'REQUIREMENT_PLOT_ADJACENT_UNIT_TYPE_MATCHES',
    PLOT_BIOME_TYPE_MATCHES: 'REQUIREMENT_PLOT_BIOME_TYPE_MATCHES',
    PLOT_DISTRICT_CLASS: 'REQUIREMENT_PLOT_DISTRICT_CLASS',
    PLOT_DISTRICT_IS_DEFENDED: 'REQUIREMENT_PLOT_DISTRICT_IS_DEFENDED',
    PLOT_FEATURE_TAG_MATCHES: 'REQUIREMENT_PLOT_FEATURE_TAG_MATCHES',
    PLOT_FEATURE_TYPE_MATCHES: 'REQUIREMENT_PLOT_FEATURE_TYPE_MATCHES',
    PLOT_HAS_ANY_DISTRICT: 'REQUIREMENT_PLOT_HAS_ANY_DISTRICT',
    PLOT_HAS_ANY_FEATURE: 'REQUIREMENT_PLOT_HAS_ANY_FEATURE',
    PLOT_HAS_ANY_IMPROVEMENT: 'REQUIREMENT_PLOT_HAS_ANY_IMPROVEMENT',
    PLOT_HAS_CONSTRUCTIBLE: 'REQUIREMENT_PLOT_HAS_CONSTRUCTIBLE',
    PLOT_HAS_EFFECT: 'REQUIREMENT_PLOT_HAS_EFFECT',
    PLOT_HAS_NUM_CONSTRUCTIBLES: 'REQUIREMENT_PLOT_HAS_NUM_CONSTRUCTIBLES',
    PLOT_HAS_ROUTE: 'REQUIREMENT_PLOT_HAS_ROUTE',
    PLOT_HAS_ROUTE_TYPE: 'REQUIREMENT_PLOT_HAS_ROUTE_TYPE',
    PLOT_HAS_YIELD: 'REQUIREMENT_PLOT_HAS_YIELD',
    PLOT_IN_COMMAND_RADIUS: 'REQUIREMENT_PLOT_IN_COMMAND_RADIUS',
    PLOT_IS_COAST: 'REQUIREMENT_PLOT_IS_COAST',
    PLOT_IS_COASTAL_LAND: 'REQUIREMENT_PLOT_IS_COASTAL_LAND',
    PLOT_IS_FRESH_WATER: 'REQUIREMENT_PLOT_IS_FRESH_WATER',
    PLOT_IS_HILLS: 'REQUIREMENT_PLOT_IS_HILLS',
    PLOT_IS_HOMELANDS: 'REQUIREMENT_PLOT_IS_HOMELANDS',
    PLOT_IS_LAKE: 'REQUIREMENT_PLOT_IS_LAKE',
    PLOT_IS_NATURAL_WONDER: 'REQUIREMENT_PLOT_IS_NATURAL_WONDER',
    PLOT_IS_OWNER: 'REQUIREMENT_PLOT_IS_OWNER',
    PLOT_IS_OWNER_CAPITAL_CONTINENT: 'REQUIREMENT_PLOT_IS_OWNER_CAPITAL_CONTINENT',
    PLOT_IS_OWNER_CAPITAL_HEMISPHERE: 'REQUIREMENT_PLOT_IS_OWNER_CAPITAL_HEMISPHERE',
    PLOT_IS_QUARTER: 'REQUIREMENT_PLOT_IS_QUARTER',
    PLOT_IS_RIVER: 'REQUIREMENT_PLOT_IS_RIVER',
    PLOT_IS_SUZERAIN: 'REQUIREMENT_PLOT_IS_SUZERAIN',
    PLOT_NEAR_CAPITAL: 'REQUIREMENT_PLOT_NEAR_CAPITAL',
    PLOT_PROPERTY_MATCHES: 'REQUIREMENT_PLOT_PROPERTY_MATCHES',
    PLOT_RESOURCE_CLASS_TYPE_MATCHES: 'REQUIREMENT_PLOT_RESOURCE_CLASS_TYPE_MATCHES',
    PLOT_RESOURCE_TAG_MATCHES: 'REQUIREMENT_PLOT_RESOURCE_TAG_MATCHES',
    PLOT_RESOURCE_TYPE_MATCHES: 'REQUIREMENT_PLOT_RESOURCE_TYPE_MATCHES',
    PLOT_RESOURCE_VISIBLE: 'REQUIREMENT_PLOT_RESOURCE_VISIBLE',
    PLOT_TERRAIN_TYPE_MATCHES: 'REQUIREMENT_PLOT_TERRAIN_TYPE_MATCHES',
    PLOT_UNIQUE_QUARTER: 'REQUIREMENT_PLOT_UNIQUE_QUARTER',
    PLOT_UNIT_HOMELANDS: 'REQUIREMENT_PLOT_UNIT_HOMELANDS',
    PLOT_UNIT_TYPE_MATCHES: 'REQUIREMENT_PLOT_UNIT_TYPE_MATCHES',
    RELIGION_IS_MAJORITY: 'REQUIREMENT_RELIGION_IS_MAJORITY',
    REQUIREMENTSET_IS_MET: 'REQUIREMENT_REQUIREMENTSET_IS_MET',
    RESIDENT_EMBASSY: 'REQUIREMENT_RESIDENT_EMBASSY',
    SPECIFIC_LEADER_ELIMINATED: 'REQUIREMENT_SPECIFIC_LEADER_ELIMINATED',
    TARGET_IS_ADJACENT_TO_FRIENDLY_UNIT: 'REQUIREMENT_TARGET_IS_ADJACENT_TO_FRIENDLY_UNIT',
    TEAM_CONQUERED_X_CITY_STATES: 'REQUIREMENT_TEAM_CONQUERED_X_CITY_STATES',
    TEAM_DOMINATION_VICTORY: 'REQUIREMENT_TEAM_DOMINATION_VICTORY',
    TEAM_HAS_HIGHEST_SCORE: 'REQUIREMENT_TEAM_HAS_HIGHEST_SCORE',
    TEAM_HAS_MOST_RELIGION_FOLLOWERS: 'REQUIREMENT_TEAM_HAS_MOST_RELIGION_FOLLOWERS',
    TEAM_HAS_MOST_TERRITORY: 'REQUIREMENT_TEAM_HAS_MOST_TERRITORY',
    TEAM_HAS_MOST_WMD_CITY_HITS: 'REQUIREMENT_TEAM_HAS_MOST_WMD_CITY_HITS',
    TEAM_LEGACY_VICTORY: 'REQUIREMENT_TEAM_LEGACY_VICTORY',
    UNIT_ADJACENT_TO_OWNER_TERRITORY: 'REQUIREMENT_UNIT_ADJACENT_TO_OWNER_TERRITORY',
    UNIT_ARMY_HAS_UNITS: 'REQUIREMENT_UNIT_ARMY_HAS_UNITS',
    UNIT_ARMY_REINFORCEMENT_STARTED: 'REQUIREMENT_UNIT_ARMY_REINFORCEMENT_STARTED',
    UNIT_CLASS_MATCHES: 'REQUIREMENT_UNIT_CLASS_MATCHES',
    UNIT_CORE_CLASS_MATCHES: 'REQUIREMENT_UNIT_CORE_CLASS_MATCHES',
    UNIT_DAMAGE_MINIMUM: 'REQUIREMENT_UNIT_DAMAGE_MINIMUM',
    UNIT_DOMAIN_MATCHES: 'REQUIREMENT_UNIT_DOMAIN_MATCHES',
    UNIT_ERA_TYPE_MATCHES: 'REQUIREMENT_UNIT_ERA_TYPE_MATCHES',
    UNIT_FOCUSED_ATTACK: 'REQUIREMENT_UNIT_FOCUSED_ATTACK',
    UNIT_FORMATION_CLASS_MATCHES: 'REQUIREMENT_UNIT_FORMATION_CLASS_MATCHES',
    UNIT_HAS_ABILITY: 'REQUIREMENT_UNIT_HAS_ABILITY',
    UNIT_HAS_BEEN_CREATED_BY_PLAYER: 'REQUIREMENT_UNIT_HAS_BEEN_CREATED_BY_PLAYER',
    UNIT_HAS_BEEN_KILLED_BY_PLAYER: 'REQUIREMENT_UNIT_HAS_BEEN_KILLED_BY_PLAYER',
    UNIT_HAS_PACKED_UNITS: 'REQUIREMENT_UNIT_HAS_PACKED_UNITS',
    UNIT_HAS_PACKED_UNITS_AFTER_TURN: 'REQUIREMENT_UNIT_HAS_PACKED_UNITS_AFTER_TURN',
    UNIT_HAS_PACKED_UNITS_BY_TAGS_OR_CORE_CLASSES: 'REQUIREMENT_UNIT_HAS_PACKED_UNITS_BY_TAGS_OR_CORE_CLASSES',
    UNIT_HAS_XP: 'REQUIREMENT_UNIT_HAS_XP',
    UNIT_HEALTH_THRESHOLD: 'REQUIREMENT_UNIT_HEALTH_THRESHOLD',
    UNIT_IN_ARMY: 'REQUIREMENT_UNIT_IN_ARMY',
    UNIT_IN_ENEMY_TERRITORY: 'REQUIREMENT_UNIT_IN_ENEMY_TERRITORY',
    UNIT_IN_FORMATION: 'REQUIREMENT_UNIT_IN_FORMATION',
    UNIT_IN_OWNER_TERRITORY: 'REQUIREMENT_UNIT_IN_OWNER_TERRITORY',
    UNIT_IS_BARBARIAN: 'REQUIREMENT_UNIT_IS_BARBARIAN',
    UNIT_IS_COMBAT: 'REQUIREMENT_UNIT_IS_COMBAT',
    UNIT_IS_ENEMY_OF_OWNER: 'REQUIREMENT_UNIT_IS_ENEMY_OF_OWNER',
    UNIT_IS_FORTIFIED: 'REQUIREMENT_UNIT_IS_FORTIFIED',
    UNIT_IS_IN_HOMELANDS: 'REQUIREMENT_UNIT_IS_IN_HOMELANDS',
    UNIT_IS_INDEPENDENT: 'REQUIREMENT_UNIT_IS_INDEPENDENT',
    UNIT_IS_STATIONED_ON_DISTRICT: 'REQUIREMENT_UNIT_IS_STATIONED_ON_DISTRICT',
    UNIT_IS_WOUNDED: 'REQUIREMENT_UNIT_IS_WOUNDED',
    UNIT_NEAR_CITY_WITH_PROJECT: 'REQUIREMENT_UNIT_NEAR_CITY_WITH_PROJECT',
    UNIT_ON_COAST: 'REQUIREMENT_UNIT_ON_COAST',
    UNIT_ON_DISTRICT: 'REQUIREMENT_UNIT_ON_DISTRICT',
    UNIT_ON_HOME_CONTINENT: 'REQUIREMENT_UNIT_ON_HOME_CONTINENT',
    UNIT_ORIGIN_HAS_SPECIFIC_CONSTRUCTIBLE: 'REQUIREMENT_UNIT_ORIGIN_HAS_SPECIFIC_CONSTRUCTIBLE',
    UNIT_TAG_MATCHES: 'REQUIREMENT_UNIT_TAG_MATCHES',
    UNIT_TRADE_ROUTE_ESTABLISHED: 'REQUIREMENT_UNIT_TRADE_ROUTE_ESTABLISHED',
    UNIT_TYPE_MATCHES: 'REQUIREMENT_UNIT_TYPE_MATCHES',
    WONDER_IS_ACTIVE: 'REQUIREMENT_WONDER_IS_ACTIVE',
} as const;
