// civ7-modding-tools@1.3.0/dist/constants/EFFECT.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.EFFECT = void 0;
exports.EFFECT = {
    ADD_BELIEF: 'EFFECT_ADD_BELIEF',
    ADD_EVENT_FREQUENCY_OVERRIDE: 'EFFECT_ADD_EVENT_FREQUENCY_OVERRIDE',
    ADD_PANTHEON: 'EFFECT_ADD_PANTHEON',
    ADD_PLAYER_BELIEF_YIELD: 'EFFECT_ADD_PLAYER_BELIEF_YIELD',
    ADD_PLAYER_UNITS_PILLAGE_BUILDING_PLUNDER: 'EFFECT_ADD_PLAYER_UNITS_PILLAGE_BUILDING_PLUNDER',
    ADD_PLAYER_UNITS_PILLAGE_IMPROVEMENT_PLUNDER: 'EFFECT_ADD_PLAYER_UNITS_PILLAGE_IMPROVEMENT_PLUNDER',
    ADD_RELIGIOUS_BELIEF_YIELD: 'EFFECT_ADD_RELIGIOUS_BELIEF_YIELD',
    ADD_RELIGIOUS_BUILDING: 'EFFECT_ADD_RELIGIOUS_BUILDING',
    ADD_RELIGIOUS_BUILDING_MULTIPLIER: 'EFFECT_ADD_RELIGIOUS_BUILDING_MULTIPLIER',
    ADD_RELIGIOUS_UNIT: 'EFFECT_ADD_RELIGIOUS_UNIT',
    ADJUST_BUILDING_SPREAD_CHARGES: 'EFFECT_ADJUST_BUILDING_SPREAD_CHARGES',
    ADJUST_CITIES_HAS_URBAN_DEFENSES: 'EFFECT_ADJUST_CITIES_HAS_URBAN_DEFENSES',
    ADJUST_CITY_AUTO_TREASURE_FLEET: 'EFFECT_ADJUST_CITY_AUTO_TREASURE_FLEET',
    ADJUST_CITY_COMMANDER_UNHAPPINESS_REDUCTION: 'EFFECT_ADJUST_CITY_COMMANDER_UNHAPPINESS_REDUCTION',
    ADJUST_CITY_IGNORE_UNHAPPINESS_EFFECT: 'EFFECT_ADJUST_CITY_IGNORE_UNHAPPINESS_EFFECT',
    ADJUST_CITY_RELIGION_EXTRA_PROMOTIONS: 'EFFECT_ADJUST_CITY_RELIGION_EXTRA_PROMOTIONS',
    ADJUST_CITY_RELIGION_ON_CAPTURE: 'EFFECT_ADJUST_CITY_RELIGION_ON_CAPTURE',
    ADJUST_CITY_TREASURE_FLEET_MOD: 'EFFECT_ADJUST_CITY_TREASURE_FLEET_MOD',
    ADJUST_CITY_UNIT_MAX_LEVEL: 'EFFECT_ADJUST_CITY_UNIT_MAX_LEVEL',
    ADJUST_DISTRICT_BASE_YIELD_CHANGE: 'EFFECT_ADJUST_DISTRICT_BASE_YIELD_CHANGE',
    ADJUST_DISTRICT_YIELD_CHANGE: 'EFFECT_ADJUST_DISTRICT_YIELD_CHANGE',
    ADJUST_DISTRICT_YIELD_MODIFIER: 'EFFECT_ADJUST_DISTRICT_YIELD_MODIFIER',
    ADJUST_FORBID_UPGRADE_TOWN_TO_CITY: 'EFFECT_ADJUST_FORBID_UPGRADE_TOWN_TO_CITY',
    ADJUST_GAINS_ALL_FOLLOWER_BELIEFS: 'EFFECT_ADJUST_GAINS_ALL_FOLLOWER_BELIEFS',
    ADJUST_GREAT_PEOPLE_POINTS_PER_KILL: 'EFFECT_ADJUST_GREAT_PEOPLE_POINTS_PER_KILL',
    ADJUST_GREAT_PERSON_GUARANTEE: 'EFFECT_ADJUST_GREAT_PERSON_GUARANTEE',
    ADJUST_GREAT_PERSON_PATRONAGE_DISCOUNT_PERCENT: 'EFFECT_ADJUST_GREAT_PERSON_PATRONAGE_DISCOUNT_PERCENT',
    ADJUST_GREAT_PERSON_POINTS: 'EFFECT_ADJUST_GREAT_PERSON_POINTS',
    ADJUST_GREAT_PERSON_POINTS_PERCENT: 'EFFECT_ADJUST_GREAT_PERSON_POINTS_PERCENT',
    ADJUST_GREAT_PERSON_POINTS_REFUND_PERCENT: 'EFFECT_ADJUST_GREAT_PERSON_POINTS_REFUND_PERCENT',
    ADJUST_HEAL_CHARGES: 'EFFECT_ADJUST_HEAL_CHARGES',
    ADJUST_MULTIPLY_TREASURY: 'EFFECT_ADJUST_MULTIPLY_TREASURY',
    ADJUST_PLAYER_ALLIANCE_CAPITAL_ROUTES: 'EFFECT_ADJUST_PLAYER_ALLIANCE_CAPITAL_ROUTES',
    ADJUST_PLAYER_ALLIANCE_TRADE: 'EFFECT_ADJUST_PLAYER_ALLIANCE_TRADE',
    ADJUST_PLAYER_CITY_TILES: 'EFFECT_ADJUST_PLAYER_CITY_TILES',
    ADJUST_PLAYER_DISTRICT_AIR_SLOTS: 'EFFECT_ADJUST_PLAYER_DISTRICT_AIR_SLOTS',
    ADJUST_PLAYER_ENFORCE_BORDERS: 'EFFECT_ADJUST_PLAYER_ENFORCE_BORDERS',
    ADJUST_PLAYER_FREE_CIVIC_ON_CITY_STATE: 'EFFECT_ADJUST_PLAYER_FREE_CIVIC_ON_CITY_STATE',
    ADJUST_PLAYER_FREE_GREAT_PERSON_POINTS: 'EFFECT_ADJUST_PLAYER_FREE_GREAT_PERSON_POINTS',
    ADJUST_PLAYER_FREE_GREAT_WORK_ON_CITY_STATE: 'EFFECT_ADJUST_PLAYER_FREE_GREAT_WORK_ON_CITY_STATE',
    ADJUST_PLAYER_FREE_POLPULATION_CAPITAL_ON_CITY_STATE: 'EFFECT_ADJUST_PLAYER_FREE_POLPULATION_CAPITAL_ON_CITY_STATE',
    ADJUST_PLAYER_FREE_TECH_ON_CITY_STATE: 'EFFECT_ADJUST_PLAYER_FREE_TECH_ON_CITY_STATE',
    ADJUST_PLAYER_JOINTWAR_PLUNDER: 'EFFECT_ADJUST_PLAYER_JOINTWAR_PLUNDER',
    ADJUST_PLAYER_LEVY_DISCOUNT_PERCENT: 'EFFECT_ADJUST_PLAYER_LEVY_DISCOUNT_PERCENT',
    ADJUST_PLAYER_NO_OCCUPATION_PENALTIES: 'EFFECT_ADJUST_PLAYER_NO_OCCUPATION_PENALTIES',
    ADJUST_PLAYER_OPEN_BORDERS_FROM_INFLUENCE: 'EFFECT_ADJUST_PLAYER_OPEN_BORDERS_FROM_INFLUENCE',
    ADJUST_PLAYER_OR_CITY_ALL_PURCHASE_EFFICIENCY: 'EFFECT_ADJUST_PLAYER_OR_CITY_ALL_PURCHASE_EFFICIENCY',
    ADJUST_PLAYER_OR_CITY_BUILDING_PURCHASE_EFFICIENCY: 'EFFECT_ADJUST_PLAYER_OR_CITY_BUILDING_PURCHASE_EFFICIENCY',
    ADJUST_PLAYER_OR_CITY_UNIT_PURCHASE_EFFICIENCY: 'EFFECT_ADJUST_PLAYER_OR_CITY_UNIT_PURCHASE_EFFICIENCY',
    ADJUST_PLAYER_RELIC_CONVERTING_CAPITAL: 'EFFECT_ADJUST_PLAYER_RELIC_CONVERTING_CAPITAL',
    ADJUST_PLAYER_RELIC_CONVERTING_CITY_STATE: 'EFFECT_ADJUST_PLAYER_RELIC_CONVERTING_CITY_STATE',
    ADJUST_PLAYER_RELIC_CONVERTING_RELIGIOUS_BUILDING: 'EFFECT_ADJUST_PLAYER_RELIC_CONVERTING_RELIGIOUS_BUILDING',
    ADJUST_PLAYER_RELIC_CONVERTING_RURAL_POP: 'EFFECT_ADJUST_PLAYER_RELIC_CONVERTING_RURAL_POP',
    ADJUST_PLAYER_RELIC_CONVERTING_TREASURE_FLEETS: 'EFFECT_ADJUST_PLAYER_RELIC_CONVERTING_TREASURE_FLEETS',
    ADJUST_PLAYER_RELIC_CONVERTING_URBAN_POP: 'EFFECT_ADJUST_PLAYER_RELIC_CONVERTING_URBAN_POP',
    ADJUST_PLAYER_RELIC_CONVERTING_WONDER: 'EFFECT_ADJUST_PLAYER_RELIC_CONVERTING_WONDER',
    ADJUST_PLAYER_RELIC_FOR_BUILDING_WONDER: 'EFFECT_ADJUST_PLAYER_RELIC_FOR_BUILDING_WONDER',
    ADJUST_PLAYER_RELIC_FOR_GOLDEN_AGE: 'EFFECT_ADJUST_PLAYER_RELIC_FOR_GOLDEN_AGE',
    ADJUST_PLAYER_RELIC_FOR_PROGRESSION_TREE_MASTERY: 'EFFECT_ADJUST_PLAYER_RELIC_FOR_PROGRESSION_TREE_MASTERY',
    ADJUST_PLAYER_RELIC_FOR_RESOURCE: 'EFFECT_ADJUST_PLAYER_RELIC_FOR_RESOURCE',
    ADJUST_PLAYER_RELIC_NEW_WORLD: 'EFFECT_ADJUST_PLAYER_RELIC_NEW_WORLD',
    ADJUST_PLAYER_SPY_BONUS: 'EFFECT_ADJUST_PLAYER_SPY_BONUS',
    ADJUST_PLAYER_STEAL_TECH_BOOSTS: 'EFFECT_ADJUST_PLAYER_STEAL_TECH_BOOSTS',
    ADJUST_PLAYER_TRADE_DURING_WAR: 'EFFECT_ADJUST_PLAYER_TRADE_DURING_WAR',
    ADJUST_PLAYER_TRADE_UNITS_SAFE_DEEP_WATER: 'EFFECT_ADJUST_PLAYER_TRADE_UNITS_SAFE_DEEP_WATER',
    ADJUST_PLAYER_UNIT_DISTRICT_PERCENT: 'EFFECT_ADJUST_PLAYER_UNIT_DISTRICT_PERCENT',
    ADJUST_PLAYER_UNIT_WONDER_PERCENT: 'EFFECT_ADJUST_PLAYER_UNIT_WONDER_PERCENT',
    ADJUST_PLAYER_UNITS_PILLAGE_BUILDING_MODIFIER: 'EFFECT_ADJUST_PLAYER_UNITS_PILLAGE_BUILDING_MODIFIER',
    ADJUST_PLAYER_UNITS_PILLAGE_IMPROVEMENT_MODIFIER: 'EFFECT_ADJUST_PLAYER_UNITS_PILLAGE_IMPROVEMENT_MODIFIER',
    ADJUST_PLAYER_VALID_IMPROVEMENT: 'EFFECT_ADJUST_PLAYER_VALID_IMPROVEMENT',
    ADJUST_PLAYER_VALID_UNIT_BUILD: 'EFFECT_ADJUST_PLAYER_VALID_UNIT_BUILD',
    ADJUST_PLAYER_WMD_COUNT: 'EFFECT_ADJUST_PLAYER_WMD_COUNT',
    ADJUST_PLAYER_YIELD_FOR_RESOURCE: 'EFFECT_ADJUST_PLAYER_YIELD_FOR_RESOURCE',
    ADJUST_PLAYER_YIELD_MODIFIER_PER_EARNED_GREAT_PERSON: 'EFFECT_ADJUST_PLAYER_YIELD_MODIFIER_PER_EARNED_GREAT_PERSON',
    ADJUST_PLAYER_YIELD_PER_SLOTTED_RESOURCE: 'EFFECT_ADJUST_PLAYER_YIELD_PER_SLOTTED_RESOURCE',
    ADJUST_PLOT_PURCHASE_COST: 'EFFECT_ADJUST_PLOT_PURCHASE_COST',
    ADJUST_RELIGIOUS_COMBAT_LOSS: 'EFFECT_ADJUST_RELIGIOUS_COMBAT_LOSS',
    ADJUST_RELIGIOUS_SPREAD_DISTANCE: 'EFFECT_ADJUST_RELIGIOUS_SPREAD_DISTANCE',
    ADJUST_RELIGIOUS_SPREAD_STRENGTH: 'EFFECT_ADJUST_RELIGIOUS_SPREAD_STRENGTH',
    ADJUST_TOWN_CAN_PURCHASE_TAGGED_CONSTRUCTIBLES: 'EFFECT_ADJUST_TOWN_CAN_PURCHASE_TAGGED_CONSTRUCTIBLES',
    ADJUST_UNIT_ADVANCED_PILLAGING: 'EFFECT_ADJUST_UNIT_ADVANCED_PILLAGING',
    ADJUST_UNIT_AREA_DAMAGE: 'EFFECT_ADJUST_UNIT_AREA_DAMAGE',
    ADJUST_UNIT_ATTACK_CREATES_PLOT_EFFECT: 'EFFECT_ADJUST_UNIT_ATTACK_CREATES_PLOT_EFFECT',
    ADJUST_UNIT_ATTACK_RANGE: 'EFFECT_ADJUST_UNIT_ATTACK_RANGE',
    ADJUST_UNIT_BASE_COMBAT_STRENGTH: 'EFFECT_ADJUST_UNIT_BASE_COMBAT_STRENGTH',
    ADJUST_UNIT_BOOST_ALL_SPIES: 'EFFECT_ADJUST_UNIT_BOOST_ALL_SPIES',
    ADJUST_UNIT_BUILD_CHARGES: 'EFFECT_ADJUST_UNIT_BUILD_CHARGES',
    ADJUST_UNIT_BUILDING_PILLAGE_YIELD_MODIFIER: 'EFFECT_ADJUST_UNIT_BUILDING_PILLAGE_YIELD_MODIFIER',
    ADJUST_UNIT_BYPASS_COMBAT_UNIT: 'EFFECT_ADJUST_UNIT_BYPASS_COMBAT_UNIT',
    ADJUST_UNIT_BYPASS_WALLS: 'EFFECT_ADJUST_UNIT_BYPASS_WALLS',
    ADJUST_UNIT_CANNOT_ATTACK: 'EFFECT_ADJUST_UNIT_CANNOT_ATTACK',
    ADJUST_UNIT_CAPITAL_RESOURCES_COMBAT_MODIFIER: 'EFFECT_ADJUST_UNIT_CAPITAL_RESOURCES_COMBAT_MODIFIER',
    ADJUST_UNIT_CIV_UNIQUE_TRADITION_COMBAT_MODIFIER: 'EFFECT_ADJUST_UNIT_CIV_UNIQUE_TRADITION_COMBAT_MODIFIER',
    ADJUST_UNIT_CLEAR_TERRAIN_START_MOVEMENT: 'EFFECT_ADJUST_UNIT_CLEAR_TERRAIN_START_MOVEMENT',
    ADJUST_UNIT_COMBAT_CAPTURE: 'EFFECT_ADJUST_UNIT_COMBAT_CAPTURE',
    ADJUST_UNIT_COMBAT_STRENGTH_MULTIPLE_ADJACENT_ENEMIES_MODIFIER: 'EFFECT_ADJUST_UNIT_COMBAT_STRENGTH_MULTIPLE_ADJACENT_ENEMIES_MODIFIER',
    ADJUST_UNIT_COMBAT_STRENGTH_PREVIOUS_ENGAGEMENT_KILL_BONUS: 'EFFECT_ADJUST_UNIT_COMBAT_STRENGTH_PREVIOUS_ENGAGEMENT_KILL_BONUS',
    ADJUST_UNIT_CONVERT_INDEPENDENT_CHARGES: 'EFFECT_ADJUST_UNIT_CONVERT_INDEPENDENT_CHARGES',
    ADJUST_UNIT_CONVERTS_BARBARIANS: 'EFFECT_ADJUST_UNIT_CONVERTS_BARBARIANS',
    ADJUST_UNIT_EMBARKED_MOVEMENT: 'EFFECT_ADJUST_UNIT_EMBARKED_MOVEMENT',
    ADJUST_UNIT_ENABLE_WALL_ATTACK: 'EFFECT_ADJUST_UNIT_ENABLE_WALL_ATTACK',
    ADJUST_UNIT_ENABLE_WALL_ATTACK_WHOLE_GAME_SAME_RELIGION: 'EFFECT_ADJUST_UNIT_ENABLE_WALL_ATTACK_WHOLE_GAME_SAME_RELIGION',
    ADJUST_UNIT_ENTER_FOREIGN_LANDS: 'EFFECT_ADJUST_UNIT_ENTER_FOREIGN_LANDS',
    ADJUST_UNIT_ESCAPE_BOOST: 'EFFECT_ADJUST_UNIT_ESCAPE_BOOST',
    ADJUST_UNIT_EVICT_PERCENT: 'EFFECT_ADJUST_UNIT_EVICT_PERCENT',
    ADJUST_UNIT_EXERT_ZOC: 'EFFECT_ADJUST_UNIT_EXERT_ZOC',
    ADJUST_UNIT_FOCUSED_ATTACK_DAMAGE: 'EFFECT_ADJUST_UNIT_FOCUSED_ATTACK_DAMAGE',
    ADJUST_UNIT_FORCE_RETREAT: 'EFFECT_ADJUST_UNIT_FORCE_RETREAT',
    ADJUST_UNIT_FOREIGN_SPREAD_MODIFIER: 'EFFECT_ADJUST_UNIT_FOREIGN_SPREAD_MODIFIER',
    ADJUST_UNIT_FRIENDLY_TERRITORY_COMBAT: 'EFFECT_ADJUST_UNIT_FRIENDLY_TERRITORY_COMBAT',
    ADJUST_UNIT_FRIENDLY_TERRITORY_START_MOVEMENT: 'EFFECT_ADJUST_UNIT_FRIENDLY_TERRITORY_START_MOVEMENT',
    ADJUST_UNIT_GREAT_PERSON_CHARGES: 'EFFECT_ADJUST_UNIT_GREAT_PERSON_CHARGES',
    ADJUST_UNIT_GREATWORKS_COMBAT_MODIFIER: 'EFFECT_ADJUST_UNIT_GREATWORKS_COMBAT_MODIFIER',
    ADJUST_UNIT_HIDDEN_VISIBILITY: 'EFFECT_ADJUST_UNIT_HIDDEN_VISIBILITY',
    ADJUST_UNIT_HOLY_CITIES_COMBAT_MODIFIER: 'EFFECT_ADJUST_UNIT_HOLY_CITIES_COMBAT_MODIFIER',
    ADJUST_UNIT_IGNORE_RESOURCE_MAINTENANCE: 'EFFECT_ADJUST_UNIT_IGNORE_RESOURCE_MAINTENANCE',
    ADJUST_UNIT_IGNORE_ZOC: 'EFFECT_ADJUST_UNIT_IGNORE_ZOC',
    ADJUST_UNIT_IMPROVEMENT_PILLAGE_YIELD_MODIFIER: 'EFFECT_ADJUST_UNIT_IMPROVEMENT_PILLAGE_YIELD_MODIFIER',
    ADJUST_UNIT_INITIATION_YIELD: 'EFFECT_ADJUST_UNIT_INITIATION_YIELD',
    ADJUST_UNIT_LAND_VICTORY_SPREAD: 'EFFECT_ADJUST_UNIT_LAND_VICTORY_SPREAD',
    ADJUST_UNIT_NATURAL_WONDER_DEFERRED_CHARGES: 'EFFECT_ADJUST_UNIT_NATURAL_WONDER_DEFERRED_CHARGES',
    ADJUST_UNIT_NEIGHBOR_COMBAT_MODIFIER: 'EFFECT_ADJUST_UNIT_NEIGHBOR_COMBAT_MODIFIER',
    ADJUST_UNIT_NO_FOREIGN_SPREAD: 'EFFECT_ADJUST_UNIT_NO_FOREIGN_SPREAD',
    ADJUST_UNIT_NO_REDUCTION_DAMAGE: 'EFFECT_ADJUST_UNIT_NO_REDUCTION_DAMAGE',
    ADJUST_UNIT_NUM_ATTACKS: 'EFFECT_ADJUST_UNIT_NUM_ATTACKS',
    ADJUST_UNIT_OWNER: 'EFFECT_ADJUST_UNIT_OWNER',
    ADJUST_UNIT_PERCENT_PILLAGE_BUILDING_MODIFIER: 'EFFECT_ADJUST_UNIT_PERCENT_PILLAGE_BUILDING_MODIFIER',
    ADJUST_UNIT_PERCENT_PILLAGE_IMPROVEMENT_MODIFIER: 'EFFECT_ADJUST_UNIT_PERCENT_PILLAGE_IMPROVEMENT_MODIFIER',
    ADJUST_UNIT_PLUNDER_YIELDS: 'EFFECT_ADJUST_UNIT_PLUNDER_YIELDS',
    ADJUST_UNIT_POST_COMBAT_HEAL: 'EFFECT_ADJUST_UNIT_POST_COMBAT_HEAL',
    ADJUST_UNIT_POST_COMBAT_YIELD: 'EFFECT_ADJUST_UNIT_POST_COMBAT_YIELD',
    ADJUST_UNIT_RAIDING: 'EFFECT_ADJUST_UNIT_RAIDING',
    ADJUST_UNIT_RELATIONSHIP_COMBAT_STRENGTH: 'EFFECT_ADJUST_UNIT_RELATIONSHIP_COMBAT_STRENGTH',
    ADJUST_UNIT_RELIC_UPON_DEATH: 'EFFECT_ADJUST_UNIT_RELIC_UPON_DEATH',
    ADJUST_UNIT_RESOURCE_DAMAGE: 'EFFECT_ADJUST_UNIT_RESOURCE_DAMAGE',
    ADJUST_UNIT_ROUGH_COMBAT_IGNORE_MODIFIER: 'EFFECT_ADJUST_UNIT_ROUGH_COMBAT_IGNORE_MODIFIER',
    ADJUST_UNIT_SEE_HIDDEN: 'EFFECT_ADJUST_UNIT_SEE_HIDDEN',
    ADJUST_UNIT_SEE_THROUGH_FEATURES: 'EFFECT_ADJUST_UNIT_SEE_THROUGH_FEATURES',
    ADJUST_UNIT_SEE_THROUGH_TERRAIN: 'EFFECT_ADJUST_UNIT_SEE_THROUGH_TERRAIN',
    ADJUST_UNIT_SIGHT: 'EFFECT_ADJUST_UNIT_SIGHT',
    ADJUST_UNIT_SPLASH_DAMAGE: 'EFFECT_ADJUST_UNIT_SPLASH_DAMAGE',
    ADJUST_UNIT_SPREAD_CHARGES: 'EFFECT_ADJUST_UNIT_SPREAD_CHARGES',
    ADJUST_UNIT_SPY_ESTABLISH_TIME: 'EFFECT_ADJUST_UNIT_SPY_ESTABLISH_TIME',
    ADJUST_UNIT_SPY_OPERATION_CHANCE: 'EFFECT_ADJUST_UNIT_SPY_OPERATION_CHANCE',
    ADJUST_UNIT_SPY_OPERATION_TIME: 'EFFECT_ADJUST_UNIT_SPY_OPERATION_TIME',
    ADJUST_UNIT_STRENGTH_MODIFIER: 'EFFECT_ADJUST_UNIT_STRENGTH_MODIFIER',
    ADJUST_UNIT_STRENGTH_PER_ENEMY_IN_COMMAND_RADIUS: 'EFFECT_ADJUST_UNIT_STRENGTH_PER_ENEMY_IN_COMMAND_RADIUS',
    ADJUST_UNIT_STRENGTH_REDUCTION_FOR_DAMAGE_MODIFIER: 'EFFECT_ADJUST_UNIT_STRENGTH_REDUCTION_FOR_DAMAGE_MODIFIER',
    ADJUST_UNIT_SUZERAIN_OF_COMBAT_MODIFIER: 'EFFECT_ADJUST_UNIT_SUZERAIN_OF_COMBAT_MODIFIER',
    ADJUST_UNIT_TRADE_ROUTE_COMBAT_MODIFIER: 'EFFECT_ADJUST_UNIT_TRADE_ROUTE_COMBAT_MODIFIER',
    ADJUST_UNIT_UNDER_SETTLEMENT_CAP_COMBAT_MODIFIER: 'EFFECT_ADJUST_UNIT_UNDER_SETTLEMENT_CAP_COMBAT_MODIFIER',
    ADJUST_UNIT_URBAN_POPULATION_COMBAT_MODIFIER: 'EFFECT_ADJUST_UNIT_URBAN_POPULATION_COMBAT_MODIFIER',
    ADJUST_UNIT_VALID_TERRAIN: 'EFFECT_ADJUST_UNIT_VALID_TERRAIN',
    ADJUST_UNIT_WAR_SUPPORT_COMBAT_STRENGTH: 'EFFECT_ADJUST_UNIT_WAR_SUPPORT_COMBAT_STRENGTH',
    ADJUST_UNIT_WATER_DAMAGE_PROTECTION: 'EFFECT_ADJUST_UNIT_WATER_DAMAGE_PROTECTION',
    ADJUST_UNIT_WMD_PROTECTION: 'EFFECT_ADJUST_UNIT_WMD_PROTECTION',
    ADJUST_VALID_FEATURES_WONDERS: 'EFFECT_ADJUST_VALID_FEATURES_WONDERS',
    ADJUST_WAR_SUPPORT_BONUS: 'EFFECT_ADJUST_WAR_SUPPORT_BONUS',
    ADJUST_WAR_WEARINESS_YIELD_PENALTY: 'EFFECT_ADJUST_WAR_WEARINESS_YIELD_PENALTY',
    ADJUST_WONDER_YIELD_CHANGE: 'EFFECT_ADJUST_WONDER_YIELD_CHANGE',
    ADJUST_YIELD_PER_ACTIVE_ENVOY: 'EFFECT_ADJUST_YIELD_PER_ACTIVE_ENVOY',
    ADJUST_YIELD_PER_ENVOY: 'EFFECT_ADJUST_YIELD_PER_ENVOY',
    ADOPT_ALLY_FOUNDED_RELIGIONS: 'EFFECT_ADOPT_ALLY_FOUNDED_RELIGIONS',
    ADVANCED_START_ADD_ARMY: 'EFFECT_ADVANCED_START_ADD_ARMY',
    ADVANCED_START_MEET_PLAYERS: 'EFFECT_ADVANCED_START_MEET_PLAYERS',
    AGE_TRANSITION_PLAYER_RESPAWN_COAST: 'EFFECT_AGE_TRANSITION_PLAYER_RESPAWN_COAST',
    ARMY_ADJUST_AIR_INTERCEPTION_RANGE: 'EFFECT_ARMY_ADJUST_AIR_INTERCEPTION_RANGE',
    ARMY_ADJUST_ALWAYS_FULL_MOVE: 'EFFECT_ARMY_ADJUST_ALWAYS_FULL_MOVE',
    ARMY_ADJUST_ATTRITION_DAMAGE: 'EFFECT_ARMY_ADJUST_ATTRITION_DAMAGE',
    ARMY_ADJUST_COMMAND_RADIUS: 'EFFECT_ARMY_ADJUST_COMMAND_RADIUS',
    ARMY_ADJUST_EXPERIENCE_RATE: 'EFFECT_ARMY_ADJUST_EXPERIENCE_RATE',
    ARMY_ADJUST_MOVEMENT_RATE: 'EFFECT_ARMY_ADJUST_MOVEMENT_RATE',
    ARMY_ADJUST_NAME: 'EFFECT_ARMY_ADJUST_NAME',
    ARMY_ADJUST_RESPAWN_SPEED: 'EFFECT_ARMY_ADJUST_RESPAWN_SPEED',
    ARMY_ADJUST_UNIT_CAPACITY: 'EFFECT_ARMY_ADJUST_UNIT_CAPACITY',
    ARMY_ADJUST_UNIT_REINFORCE_SPEED: 'EFFECT_ARMY_ADJUST_UNIT_REINFORCE_SPEED',
    ARMY_GRANT_UNIT: 'EFFECT_ARMY_GRANT_UNIT',
    ATTACH_MODIFIERS: 'EFFECT_ATTACH_MODIFIERS',
    BASIC_DIPLOMACY_MODS: 'EFFECT_BASIC_DIPLOMACY_MODS',
    BLOCK_DIPLOMAITC_ACTION_GROUP_FOR_X_TURNS: 'EFFECT_BLOCK_DIPLOMAITC_ACTION_GROUP_FOR_X_TURNS',
    CHANGE_INDEPENDENT_AT_LOCATION_INVESTMENT: 'EFFECT_CHANGE_INDEPENDENT_AT_LOCATION_INVESTMENT',
    CHANGE_INDEPENDENT_INVESTMENT: 'EFFECT_CHANGE_INDEPENDENT_INVESTMENT',
    CITY_ACTIVATE_CONSTRUCTIBLE_ADJACENCY: 'EFFECT_CITY_ACTIVATE_CONSTRUCTIBLE_ADJACENCY',
    CITY_ACTIVATE_CONSTRUCTIBLE_WAREHOUSE_YIELD: 'EFFECT_CITY_ACTIVATE_CONSTRUCTIBLE_WAREHOUSE_YIELD',
    CITY_ADD_FOOD_AFTER_GROWTH_EVENT: 'EFFECT_CITY_ADD_FOOD_AFTER_GROWTH_EVENT',
    CITY_ADD_RESOURCE_TO_PLOT: 'EFFECT_CITY_ADD_RESOURCE_TO_PLOT',
    CITY_ADJUST_ADJACENCY_FLAT_AMOUNT: 'EFFECT_CITY_ADJUST_ADJACENCY_FLAT_AMOUNT',
    CITY_ADJUST_AVOID_RANDOM_EVENT: 'EFFECT_CITY_ADJUST_AVOID_RANDOM_EVENT',
    CITY_ADJUST_BIOME_WONDER_PRODUCTION_PER_RESOURCE: 'EFFECT_CITY_ADJUST_BIOME_WONDER_PRODUCTION_PER_RESOURCE',
    CITY_ADJUST_BUILD_QUEUE: 'EFFECT_CITY_ADJUST_BUILD_QUEUE',
    CITY_ADJUST_BUILDING_MAINTENANCE_EFFICIENCY: 'EFFECT_CITY_ADJUST_BUILDING_MAINTENANCE_EFFICIENCY',
    CITY_ADJUST_BUILDING_PRODUCTION_PRODUCTION_PER_LEGACY_POINT: 'EFFECT_CITY_ADJUST_BUILDING_PRODUCTION_PRODUCTION_PER_LEGACY_POINT',
    CITY_ADJUST_CAN_PURCHASE_IN_UNREST: 'EFFECT_CITY_ADJUST_CAN_PURCHASE_IN_UNREST',
    CITY_ADJUST_CIVILIAN_UNIT_FOOD: 'EFFECT_CITY_ADJUST_CIVILIAN_UNIT_FOOD',
    CITY_ADJUST_CONSTRUCTIBLE_ADJACENCY: 'EFFECT_CITY_ADJUST_CONSTRUCTIBLE_ADJACENCY',
    CITY_ADJUST_CONSTRUCTIBLE_PRODUCTION: 'EFFECT_CITY_ADJUST_CONSTRUCTIBLE_PRODUCTION',
    CITY_ADJUST_CONSTRUCTIBLE_PRODUCTION_PER_RESOURCE: 'EFFECT_CITY_ADJUST_CONSTRUCTIBLE_PRODUCTION_PER_RESOURCE',
    CITY_ADJUST_CONSTRUCTIBLE_PRODUCTION_PER_SLOTTED_RESOURCE: 'EFFECT_CITY_ADJUST_CONSTRUCTIBLE_PRODUCTION_PER_SLOTTED_RESOURCE',
    CITY_ADJUST_CONSTRUCTIBLE_PRODUCTION_PER_SUZERAIN_OF: 'EFFECT_CITY_ADJUST_CONSTRUCTIBLE_PRODUCTION_PER_SUZERAIN_OF',
    CITY_ADJUST_CONSTRUCTIBLE_YIELD: 'EFFECT_CITY_ADJUST_CONSTRUCTIBLE_YIELD',
    CITY_ADJUST_FAVORED_WONDER_PRODUCTION: 'EFFECT_CITY_ADJUST_FAVORED_WONDER_PRODUCTION',
    CITY_ADJUST_FREE_CONSTRUCTIBLE_TIER: 'EFFECT_CITY_ADJUST_FREE_CONSTRUCTIBLE_TIER',
    CITY_ADJUST_GREAT_WORK_SLOTS: 'EFFECT_CITY_ADJUST_GREAT_WORK_SLOTS',
    CITY_ADJUST_GROWTH: 'EFFECT_CITY_ADJUST_GROWTH',
    CITY_ADJUST_GROWTH_PER_RESOURCE: 'EFFECT_CITY_ADJUST_GROWTH_PER_RESOURCE',
    CITY_ADJUST_GROWTH_PER_WORKER: 'EFFECT_CITY_ADJUST_GROWTH_PER_WORKER',
    CITY_ADJUST_HIT_POINTS: 'EFFECT_CITY_ADJUST_HIT_POINTS',
    CITY_ADJUST_LOCAL_UNHAPPINESS: 'EFFECT_CITY_ADJUST_LOCAL_UNHAPPINESS',
    CITY_ADJUST_OVERBUILD_PRODUCTION_MOD: 'EFFECT_CITY_ADJUST_OVERBUILD_PRODUCTION_MOD',
    CITY_ADJUST_POPULATION: 'EFFECT_CITY_ADJUST_POPULATION',
    CITY_ADJUST_PROJECT_PRODUCTION: 'EFFECT_CITY_ADJUST_PROJECT_PRODUCTION',
    CITY_ADJUST_PROJECT_PRODUCTION_PER_LEGACY_POINT: 'EFFECT_CITY_ADJUST_PROJECT_PRODUCTION_PER_LEGACY_POINT',
    CITY_ADJUST_PURCHASE_EFFICIENCY_PER_RESOURCE: 'EFFECT_CITY_ADJUST_PURCHASE_EFFICIENCY_PER_RESOURCE',
    CITY_ADJUST_RANDOM_EVENT_CONSTRUCTIBLE_YIELD: 'EFFECT_CITY_ADJUST_RANDOM_EVENT_CONSTRUCTIBLE_YIELD',
    CITY_ADJUST_RAZE_RATE: 'EFFECT_CITY_ADJUST_RAZE_RATE',
    CITY_ADJUST_RESOURCE_CAP: 'EFFECT_CITY_ADJUST_RESOURCE_CAP',
    CITY_ADJUST_RESOURCE_CAP_PER_RESOURCE: 'EFFECT_CITY_ADJUST_RESOURCE_CAP_PER_RESOURCE',
    CITY_ADJUST_RESOURCE_CAP_PER_SUZERAIN: 'EFFECT_CITY_ADJUST_RESOURCE_CAP_PER_SUZERAIN',
    CITY_ADJUST_RESOURCE_CAP_PER_TOWN: 'EFFECT_CITY_ADJUST_RESOURCE_CAP_PER_TOWN',
    CITY_ADJUST_SPECIFIC_CONSTRUCTIBLE_MAINTENANCE: 'EFFECT_CITY_ADJUST_SPECIFIC_CONSTRUCTIBLE_MAINTENANCE',
    CITY_ADJUST_SUZERAIN_OF_CONSTRUCTIBLE_YIELD: 'EFFECT_CITY_ADJUST_SUZERAIN_OF_CONSTRUCTIBLE_YIELD',
    CITY_ADJUST_TOTAL_DISTRICT_HEALTH: 'EFFECT_CITY_ADJUST_TOTAL_DISTRICT_HEALTH',
    CITY_ADJUST_TOWN_UPGRADE_COST_MULTIPLIER: 'EFFECT_CITY_ADJUST_TOWN_UPGRADE_COST_MULTIPLIER',
    CITY_ADJUST_TOWN_UPGRADE_DISCOUNT: 'EFFECT_CITY_ADJUST_TOWN_UPGRADE_DISCOUNT',
    CITY_ADJUST_TRADE_ROUTE_RANGE: 'EFFECT_CITY_ADJUST_TRADE_ROUTE_RANGE',
    CITY_ADJUST_TRADE_ROUTE_RANGE_PER_RESOURCE: 'EFFECT_CITY_ADJUST_TRADE_ROUTE_RANGE_PER_RESOURCE',
    CITY_ADJUST_TRADE_ROUTE_RANGE_PER_SUZERAIN_OF: 'EFFECT_CITY_ADJUST_TRADE_ROUTE_RANGE_PER_SUZERAIN_OF',
    CITY_ADJUST_TRADE_YIELD: 'EFFECT_CITY_ADJUST_TRADE_YIELD',
    CITY_ADJUST_TRADE_YIELD_PER_RESOURCE: 'EFFECT_CITY_ADJUST_TRADE_YIELD_PER_RESOURCE',
    CITY_ADJUST_TREASURE_PREREQ: 'EFFECT_CITY_ADJUST_TREASURE_PREREQ',
    CITY_ADJUST_UNHAPPINESS_EFFECT: 'EFFECT_CITY_ADJUST_UNHAPPINESS_EFFECT',
    CITY_ADJUST_UNIT_DOMAIN_PRODUCTION: 'EFFECT_CITY_ADJUST_UNIT_DOMAIN_PRODUCTION',
    CITY_ADJUST_UNIT_MOVEMENT_CLASS_PRODUCTION: 'EFFECT_CITY_ADJUST_UNIT_MOVEMENT_CLASS_PRODUCTION',
    CITY_ADJUST_UNIT_PRODUCTION: 'EFFECT_CITY_ADJUST_UNIT_PRODUCTION',
    CITY_ADJUST_UNIT_PRODUCTION_MOD_PER_SETTLEMENT: 'EFFECT_CITY_ADJUST_UNIT_PRODUCTION_MOD_PER_SETTLEMENT',
    CITY_ADJUST_UNIT_PRODUCTION_PER_RESOURCE: 'EFFECT_CITY_ADJUST_UNIT_PRODUCTION_PER_RESOURCE',
    CITY_ADJUST_UNIT_PRODUCTION_PER_SLOTTED_RESOURCE: 'EFFECT_CITY_ADJUST_UNIT_PRODUCTION_PER_SLOTTED_RESOURCE',
    CITY_ADJUST_WONDER_PRODUCTION: 'EFFECT_CITY_ADJUST_WONDER_PRODUCTION',
    CITY_ADJUST_WORKER_CAP: 'EFFECT_CITY_ADJUST_WORKER_CAP',
    CITY_ADJUST_WORKER_MAINTENANCE_EFFICIENCY: 'EFFECT_CITY_ADJUST_WORKER_MAINTENANCE_EFFICIENCY',
    CITY_ADJUST_WORKER_MAINTENANCE_EFFICIENCY_PER_RESOURCE: 'EFFECT_CITY_ADJUST_WORKER_MAINTENANCE_EFFICIENCY_PER_RESOURCE',
    CITY_ADJUST_WORKER_YIELD: 'EFFECT_CITY_ADJUST_WORKER_YIELD',
    CITY_ADJUST_WORKER_YIELD_PER_ATTRIBUTE: 'EFFECT_CITY_ADJUST_WORKER_YIELD_PER_ATTRIBUTE',
    CITY_ADJUST_WORKER_YIELD_PER_WORKER_ON_TILE: 'EFFECT_CITY_ADJUST_WORKER_YIELD_PER_WORKER_ON_TILE',
    CITY_ADJUST_YIELD: 'EFFECT_CITY_ADJUST_YIELD',
    CITY_ADJUST_YIELD_CONVERSION: 'EFFECT_CITY_ADJUST_YIELD_CONVERSION',
    CITY_ADJUST_YIELD_PER_ACTIVE_TRADITION: 'EFFECT_CITY_ADJUST_YIELD_PER_ACTIVE_TRADITION',
    CITY_ADJUST_YIELD_PER_ATTRIBUTE: 'EFFECT_CITY_ADJUST_YIELD_PER_ATTRIBUTE',
    CITY_ADJUST_YIELD_PER_AVAILABLE_RESOURCE_TYPE: 'EFFECT_CITY_ADJUST_YIELD_PER_AVAILABLE_RESOURCE_TYPE',
    CITY_ADJUST_YIELD_PER_COMMANDER_APPLIED_PROMOTIONS: 'EFFECT_CITY_ADJUST_YIELD_PER_COMMANDER_APPLIED_PROMOTIONS',
    CITY_ADJUST_YIELD_PER_COMMANDER_LEVEL: 'EFFECT_CITY_ADJUST_YIELD_PER_COMMANDER_LEVEL',
    CITY_ADJUST_YIELD_PER_CONNECTED_CITY: 'EFFECT_CITY_ADJUST_YIELD_PER_CONNECTED_CITY',
    CITY_ADJUST_YIELD_PER_DISTRICT: 'EFFECT_CITY_ADJUST_YIELD_PER_DISTRICT',
    CITY_ADJUST_YIELD_PER_GREAT_WORK: 'EFFECT_CITY_ADJUST_YIELD_PER_GREAT_WORK',
    CITY_ADJUST_YIELD_PER_NUM_CAPTURED_CITIES: 'EFFECT_CITY_ADJUST_YIELD_PER_NUM_CAPTURED_CITIES',
    CITY_ADJUST_YIELD_PER_NUM_CITIES: 'EFFECT_CITY_ADJUST_YIELD_PER_NUM_CITIES',
    CITY_ADJUST_YIELD_PER_NUM_TRADE_ROUTES: 'EFFECT_CITY_ADJUST_YIELD_PER_NUM_TRADE_ROUTES',
    CITY_ADJUST_YIELD_PER_POPULATION: 'EFFECT_CITY_ADJUST_YIELD_PER_POPULATION',
    CITY_ADJUST_YIELD_PER_RESOURCE: 'EFFECT_CITY_ADJUST_YIELD_PER_RESOURCE',
    CITY_ADJUST_YIELD_PER_RESOURCE_CLASS: 'EFFECT_CITY_ADJUST_YIELD_PER_RESOURCE_CLASS',
    CITY_ADJUST_YIELD_PER_SURPLUS_HAPPINESS: 'EFFECT_CITY_ADJUST_YIELD_PER_SURPLUS_HAPPINESS',
    CITY_ADJUST_YIELD_PER_SUZERAIN: 'EFFECT_CITY_ADJUST_YIELD_PER_SUZERAIN',
    CITY_ADJUST_YIELD_PER_UNDER_SETTLEMENT_CAP: 'EFFECT_CITY_ADJUST_YIELD_PER_UNDER_SETTLEMENT_CAP',
    CITY_ADJUST_YIELD_PER_UNLOCKED_PROGRESSION_TREE_NODE: 'EFFECT_CITY_ADJUST_YIELD_PER_UNLOCKED_PROGRESSION_TREE_NODE',
    CITY_ADJUST_YIELD_WITH_DECAY: 'EFFECT_CITY_ADJUST_YIELD_WITH_DECAY',
    CITY_ADJUST_YIELDS_PER_SETTLEMENT_OVER_CAP: 'EFFECT_CITY_ADJUST_YIELDS_PER_SETTLEMENT_OVER_CAP',
    CITY_DAMAGE_CONSTRUCTIBLES: 'EFFECT_CITY_DAMAGE_CONSTRUCTIBLES',
    CITY_GRANT_FREE_BUILDING: 'EFFECT_CITY_GRANT_FREE_BUILDING',
    CITY_GRANT_GREAT_WORK: 'EFFECT_CITY_GRANT_GREAT_WORK',
    CITY_GRANT_NATURAL_YIELDS: 'EFFECT_CITY_GRANT_NATURAL_YIELDS',
    CITY_GRANT_UNIT: 'EFFECT_CITY_GRANT_UNIT',
    CITY_GRANT_WAREHOUSE_YIELD: 'EFFECT_CITY_GRANT_WAREHOUSE_YIELD',
    CITY_GRANT_YIELD: 'EFFECT_CITY_GRANT_YIELD',
    CITY_GRANT_YIELD_DISCOVERY: 'EFFECT_CITY_GRANT_YIELD_DISCOVERY',
    CITY_GRANT_YIELD_PER_POP_DEFENSE_CONSTRUCTED: 'EFFECT_CITY_GRANT_YIELD_PER_POP_DEFENSE_CONSTRUCTED',
    CITY_MAKE_OWNER_RELIGION: 'EFFECT_CITY_MAKE_OWNER_RELIGION',
    CITY_PROPERTY: 'EFFECT_CITY_PROPERTY',
    CITY_SPAWN_TRIBE_ARMY: 'EFFECT_CITY_SPAWN_TRIBE_ARMY',
    CITY_SPAWN_TRIBE_NEAR: 'EFFECT_CITY_SPAWN_TRIBE_NEAR',
    CITY_TRANSFER_OWNER: 'EFFECT_CITY_TRANSFER_OWNER',
    CITY_UNLOCK_PROJECT: 'EFFECT_CITY_UNLOCK_PROJECT',
    COMPLETE_LAND_CLAIM: 'EFFECT_COMPLETE_LAND_CLAIM',
    DAE_ADJUST_SETTLEMENT_YIELD: 'EFFECT_DAE_ADJUST_SETTLEMENT_YIELD',
    DAE_APPLY_NON_AGGRESSION_PACT: 'EFFECT_DAE_APPLY_NON_AGGRESSION_PACT',
    DAE_COMPLETE_APPLY_BEHAVIOR: 'EFFECT_DAE_COMPLETE_APPLY_BEHAVIOR',
    DAE_COMPLETE_FREE_CAPTURED_COMMANDER: 'EFFECT_DAE_COMPLETE_FREE_CAPTURED_COMMANDER',
    DAE_COMPLETE_GRANT_COOPERATIVE_YIELD: 'EFFECT_DAE_COMPLETE_GRANT_COOPERATIVE_YIELD',
    DAE_COMPLETE_GRANT_FAVORS_GRIEVANCES: 'EFFECT_DAE_COMPLETE_GRANT_FAVORS_GRIEVANCES',
    DAE_COMPLETE_IMPROVE_TRADE_RELATIONS: 'EFFECT_DAE_COMPLETE_IMPROVE_TRADE_RELATIONS',
    DAE_COOPERATIVE_ATTACH_MODIFIER: 'EFFECT_DAE_COOPERATIVE_ATTACH_MODIFIER',
    DAE_DESCRIBE_GRANT_COOPERATIVE_YIELD: 'EFFECT_DAE_DESCRIBE_GRANT_COOPERATIVE_YIELD',
    DAE_DESCRIBE_GRANT_FAVORS_GRIEVANCES: 'EFFECT_DAE_DESCRIBE_GRANT_FAVORS_GRIEVANCES',
    DAE_DESCRIBE_IMPROVE_TRADE_RELATIONS: 'EFFECT_DAE_DESCRIBE_IMPROVE_TRADE_RELATIONS',
    DAE_DESCRIBE_INDEPENDENT_AND_TARGET: 'EFFECT_DAE_DESCRIBE_INDEPENDENT_AND_TARGET',
    DAE_ESPIONAGE_ADJUST_CITY_CONSTRUCTIBLE_PRODUCTION: 'EFFECT_DAE_ESPIONAGE_ADJUST_CITY_CONSTRUCTIBLE_PRODUCTION',
    DAE_ESPIONAGE_ADJUST_CITY_PROJECT_PRODUCTION: 'EFFECT_DAE_ESPIONAGE_ADJUST_CITY_PROJECT_PRODUCTION',
    DAE_ESPIONAGE_ATTACH_MODIFIER: 'EFFECT_DAE_ESPIONAGE_ATTACH_MODIFIER',
    DAE_ESPIONAGE_BLOCK_CITY_GREAT_WORKS: 'EFFECT_DAE_ESPIONAGE_BLOCK_CITY_GREAT_WORKS',
    DAE_ESPIONAGE_CONVERT_SETTLEMENT: 'EFFECT_DAE_ESPIONAGE_CONVERT_SETTLEMENT',
    DAE_ESPIONAGE_DESTROY_TARGET_INFRASTRUCTURE: 'EFFECT_DAE_ESPIONAGE_DESTROY_TARGET_INFRASTRUCTURE',
    DAE_ESPIONAGE_DESTROY_TARGET_INFRASTRUCTURE_WITH_RESOURCE_TYPE: 'EFFECT_DAE_ESPIONAGE_DESTROY_TARGET_INFRASTRUCTURE_WITH_RESOURCE_TYPE',
    DAE_ESPIONAGE_DISABLE_SETTLEMENT_WORKERS: 'EFFECT_DAE_ESPIONAGE_DISABLE_SETTLEMENT_WORKERS',
    DAE_ESPIONAGE_GRANT_TARGET_COMMANDER_VISION: 'EFFECT_DAE_ESPIONAGE_GRANT_TARGET_COMMANDER_VISION',
    DAE_ESPIONAGE_SPAWN_TREASURE_FLEET: 'EFFECT_DAE_ESPIONAGE_SPAWN_TREASURE_FLEET',
    DAE_ESPIONAGE_STEAL_TREE_NODE: 'EFFECT_DAE_ESPIONAGE_STEAL_TREE_NODE',
    DAE_FREE_CAPTURED_COMMANDER_CHANGE_INITIATOR_GOLD: 'EFFECT_DAE_FREE_CAPTURED_COMMANDER_CHANGE_INITIATOR_GOLD',
    DAE_GRANT_COOPERATIVE_YIELD_SUPPORT_BASE_PER_TURN: 'EFFECT_DAE_GRANT_COOPERATIVE_YIELD_SUPPORT_BASE_PER_TURN',
    DAE_GRANT_YIELD_PER_RELATIONSHIP: 'EFFECT_DAE_GRANT_YIELD_PER_RELATIONSHIP',
    DAE_INCORPORATE_CITY_STATE: 'EFFECT_DAE_INCORPORATE_CITY_STATE',
    DAE_LEVY_UNIT: 'EFFECT_DAE_LEVY_UNIT',
    DAE_SEND_GOLD: 'EFFECT_DAE_SEND_GOLD',
    DAE_START_GRANT_COOPERATIVE_YIELD: 'EFFECT_DAE_START_GRANT_COOPERATIVE_YIELD',
    DAE_TARGET_ATTACH_MODIFIER: 'EFFECT_DAE_TARGET_ATTACH_MODIFIER',
    DAE_TRADE_MAP: 'EFFECT_DAE_TRADE_MAP',
    DECLARE_WAR: 'EFFECT_DECLARE_WAR',
    DIPLOMACY_ADJUST_CITY_YIELD_PER_PLAYER_RELATIONSHIP: 'EFFECT_DIPLOMACY_ADJUST_CITY_YIELD_PER_PLAYER_RELATIONSHIP',
    DIPLOMACY_ADJUST_DIPLOMATIC_ACTION_TOKEN_BONUS: 'EFFECT_DIPLOMACY_ADJUST_DIPLOMATIC_ACTION_TOKEN_BONUS',
    DIPLOMACY_ADJUST_DIPLOMATIC_ACTION_TOKEN_BONUS_PER_PLAYER_RELATIONSHIP: 'EFFECT_DIPLOMACY_ADJUST_DIPLOMATIC_ACTION_TOKEN_BONUS_PER_PLAYER_RELATIONSHIP',
    DIPLOMACY_ADJUST_DIPLOMATIC_ACTION_TYPE_EFFICIENCY: 'EFFECT_DIPLOMACY_ADJUST_DIPLOMATIC_ACTION_TYPE_EFFICIENCY',
    DIPLOMACY_ADJUST_DIPLOMATIC_ACTION_TYPE_EFFICIENCY_PER_GREAT_WORK: 'EFFECT_DIPLOMACY_ADJUST_DIPLOMATIC_ACTION_TYPE_EFFICIENCY_PER_GREAT_WORK',
    DIPLOMACY_ADJUST_DIPLOMATIC_RESPONSE_EFFICIENCY: 'EFFECT_DIPLOMACY_ADJUST_DIPLOMATIC_RESPONSE_EFFICIENCY',
    DIPLOMACY_ADJUST_INFLUENCE_FROM_GRIEVANCES: 'EFFECT_DIPLOMACY_ADJUST_INFLUENCE_FROM_GRIEVANCES',
    DIPLOMACY_ADJUST_RELATIONSHIP_GAIN_FROM_EVENT: 'EFFECT_DIPLOMACY_ADJUST_RELATIONSHIP_GAIN_FROM_EVENT',
    DIPLOMACY_ADJUST_UNIT_MAINTENANCE_PER_PLAYER_RELATIONSHIP: 'EFFECT_DIPLOMACY_ADJUST_UNIT_MAINTENANCE_PER_PLAYER_RELATIONSHIP',
    DIPLOMACY_ADJUST_UNIT_PRODUCTION_PER_PLAYER_RELATIONSHIP: 'EFFECT_DIPLOMACY_ADJUST_UNIT_PRODUCTION_PER_PLAYER_RELATIONSHIP',
    DIPLOMACY_ADJUST_YIELD_PER_PLAYER_INVOLVED_ACTION: 'EFFECT_DIPLOMACY_ADJUST_YIELD_PER_PLAYER_INVOLVED_ACTION',
    DIPLOMACY_ADJUST_YIELD_PER_PLAYER_RELATIONSHIP: 'EFFECT_DIPLOMACY_ADJUST_YIELD_PER_PLAYER_RELATIONSHIP',
    DIPLOMACY_AGENDA_ON_DISPERSE_IP: 'EFFECT_DIPLOMACY_AGENDA_ON_DISPERSE_IP',
    DIPLOMACY_AGENDA_ON_WAR_START: 'EFFECT_DIPLOMACY_AGENDA_ON_WAR_START',
    DIPLOMACY_AGENDA_TIMED_UPDATE: 'EFFECT_DIPLOMACY_AGENDA_TIMED_UPDATE',
    DIPLOMACY_BROKEN_PROMISE: 'EFFECT_DIPLOMACY_BROKEN_PROMISE',
    DIPLOMACY_FORCE_INCURSION: 'EFFECT_DIPLOMACY_FORCE_INCURSION',
    DIPLOMACY_GRANT_DIPLOMATIC_ACTION: 'EFFECT_DIPLOMACY_GRANT_DIPLOMATIC_ACTION',
    DIPLOMACY_KEPT_PROMISE: 'EFFECT_DIPLOMACY_KEPT_PROMISE',
    DIPLOMACY_NO_PROMISE: 'EFFECT_DIPLOMACY_NO_PROMISE',
    DIPLOMACY_ONE_SIDED_TRADES: 'EFFECT_DIPLOMACY_ONE_SIDED_TRADES',
    DIPLOMACY_RANDOM: 'EFFECT_DIPLOMACY_RANDOM',
    DIPLOMACY_SIMPLE_EFFECT: 'EFFECT_DIPLOMACY_SIMPLE_EFFECT',
    DIPLOMACY_THIRD_PARTY_EFFECTS: 'EFFECT_DIPLOMACY_THIRD_PARTY_EFFECTS',
    DISTRICT_ADJUST_FORTIFIED_COMBAT_STRENGTH: 'EFFECT_DISTRICT_ADJUST_FORTIFIED_COMBAT_STRENGTH',
    DISTRICT_ADJUST_HEAL_PER_TURN: 'EFFECT_DISTRICT_ADJUST_HEAL_PER_TURN',
    DISTRICT_ADJUST_MAX_SUPPLIES: 'EFFECT_DISTRICT_ADJUST_MAX_SUPPLIES',
    DISTRICT_ADJUST_TOTAL_HEALTH: 'EFFECT_DISTRICT_ADJUST_TOTAL_HEALTH',
    DISTRICT_GRANT_SUPPLIES: 'EFFECT_DISTRICT_GRANT_SUPPLIES',
    DO_NOTHING: 'EFFECT_DO_NOTHING',
    ELEVATE_INDEPENDENT: 'EFFECT_ELEVATE_INDEPENDENT',
    ENABLE_RELIGION_AUTO_SPREAD: 'EFFECT_ENABLE_RELIGION_AUTO_SPREAD',
    ENABLE_RELIGION_AUTO_SPREAD_CONQUERED: 'EFFECT_ENABLE_RELIGION_AUTO_SPREAD_CONQUERED',
    ENABLE_RELIGION_AUTO_SPREAD_FOREIGN_HEMISPHERE: 'EFFECT_ENABLE_RELIGION_AUTO_SPREAD_FOREIGN_HEMISPHERE',
    ENABLE_RELIGION_AUTO_SPREAD_TRADE_ROUTE: 'EFFECT_ENABLE_RELIGION_AUTO_SPREAD_TRADE_ROUTE',
    ENABLE_RELIGION_AWARDS_ENVOY: 'EFFECT_ENABLE_RELIGION_AWARDS_ENVOY',
    EXPLORE_ENTIRE_MAP: 'EFFECT_EXPLORE_ENTIRE_MAP',
    GAME_PROPERTY: 'EFFECT_GAME_PROPERTY',
    GRANT_AIR_SLOTS: 'EFFECT_GRANT_AIR_SLOTS',
    GRANT_ALL_PLAYERS_FAVORS_GRIEVANCES: 'EFFECT_GRANT_ALL_PLAYERS_FAVORS_GRIEVANCES',
    GRANT_BUILDING_IN_CITY: 'EFFECT_GRANT_BUILDING_IN_CITY',
    GRANT_BUILDING_IN_CITY_ADJACENT_TO_TERRAIN: 'EFFECT_GRANT_BUILDING_IN_CITY_ADJACENT_TO_TERRAIN',
    GRANT_BUILDING_IN_CITY_IGNORE: 'EFFECT_GRANT_BUILDING_IN_CITY_IGNORE',
    GRANT_CITY_FROM_CITY_STATE_PLOT: 'EFFECT_GRANT_CITY_FROM_CITY_STATE_PLOT',
    GRANT_CITY_OWNER_INFLUENCE_TOKEN_WONDER: 'EFFECT_GRANT_CITY_OWNER_INFLUENCE_TOKEN_WONDER',
    GRANT_CITY_ROAD_TO_CAPITAL: 'EFFECT_GRANT_CITY_ROAD_TO_CAPITAL',
    GRANT_CITY_YIELD_BIOME: 'EFFECT_GRANT_CITY_YIELD_BIOME',
    GRANT_COMBAT_ADJACENCY: 'EFFECT_GRANT_COMBAT_ADJACENCY',
    GRANT_DIPLOMATIC_ENVOYS: 'EFFECT_GRANT_DIPLOMATIC_ENVOYS',
    GRANT_GREAT_PERSON_CLASS_IN_CITY: 'EFFECT_GRANT_GREAT_PERSON_CLASS_IN_CITY',
    GRANT_GREAT_WORK: 'EFFECT_GRANT_GREAT_WORK',
    GRANT_PLAYER_RELIGIOUS_PRESSURE_GREAT_PERSON_ACTIVATED: 'EFFECT_GRANT_PLAYER_RELIGIOUS_PRESSURE_GREAT_PERSON_ACTIVATED',
    GRANT_PLAYER_YIELD_PERCENT_UNIT_COST: 'EFFECT_GRANT_PLAYER_YIELD_PERCENT_UNIT_COST',
    GRANT_PLOT: 'EFFECT_GRANT_PLOT',
    GRANT_PRODUCTION_IN_CITY: 'EFFECT_GRANT_PRODUCTION_IN_CITY',
    GRANT_RETURNED_CITIES_BONUS: 'EFFECT_GRANT_RETURNED_CITIES_BONUS',
    GRANT_REVEAL_TYPE: 'EFFECT_GRANT_REVEAL_TYPE',
    GRANT_SPY: 'EFFECT_GRANT_SPY',
    GRANT_STRENGTH_PER_ADJACENT_UNIT_TYPE: 'EFFECT_GRANT_STRENGTH_PER_ADJACENT_UNIT_TYPE',
    GRANT_SUPPORT_RESPONSE_BONUS: 'EFFECT_GRANT_SUPPORT_RESPONSE_BONUS',
    GRANT_SUZERAIN_UNIT_PLOT: 'EFFECT_GRANT_SUZERAIN_UNIT_PLOT',
    GRANT_UNIT_ABILITY_CHARGE: 'EFFECT_GRANT_UNIT_ABILITY_CHARGE',
    GRANT_UNIT_EXPERIENCE: 'EFFECT_GRANT_UNIT_EXPERIENCE',
    GRANT_UNIT_OF_CLASS_AND_APPLY_ABILITY: 'EFFECT_GRANT_UNIT_OF_CLASS_AND_APPLY_ABILITY',
    GRANT_UNIT_ON_SUCCESSFUL_UNDETECTED_ESPIONAGE_ACTION: 'EFFECT_GRANT_UNIT_ON_SUCCESSFUL_UNDETECTED_ESPIONAGE_ACTION',
    GRANT_UNIT_PROMOTION: 'EFFECT_GRANT_UNIT_PROMOTION',
    GRANT_UNIT_RESPAWN_CHARGE: 'EFFECT_GRANT_UNIT_RESPAWN_CHARGE',
    GRANT_UNIT_YIELD_ADJACENT_FEATURES: 'EFFECT_GRANT_UNIT_YIELD_ADJACENT_FEATURES',
    GRANT_UNIT_YIELD_ADJACENT_NATURAL_WONDERS: 'EFFECT_GRANT_UNIT_YIELD_ADJACENT_NATURAL_WONDERS',
    GRANT_UNIT_YIELD_ADJACENT_TERRAINS: 'EFFECT_GRANT_UNIT_YIELD_ADJACENT_TERRAINS',
    GRANT_UNIT_YIELD_PER_RESOURCES_IN_CITY: 'EFFECT_GRANT_UNIT_YIELD_PER_RESOURCES_IN_CITY',
    GRANT_UNIT_YIELD_PROMOTION: 'EFFECT_GRANT_UNIT_YIELD_PROMOTION',
    GRANT_UNIT_YIELD_RIVER_LENGTH: 'EFFECT_GRANT_UNIT_YIELD_RIVER_LENGTH',
    GRANT_WALLS: 'EFFECT_GRANT_WALLS',
    GRANT_YIELD_BASED_ON_CURRENT_YIELD_RATE: 'EFFECT_GRANT_YIELD_BASED_ON_CURRENT_YIELD_RATE',
    GRANT_YIELD_FOR_GREAT_WORK: 'EFFECT_GRANT_YIELD_FOR_GREAT_WORK',
    GRANT_YIELD_PER_ENVOY: 'EFFECT_GRANT_YIELD_PER_ENVOY',
    GRANT_YIELD_PER_GREAT_WORK_IN_CITY: 'EFFECT_GRANT_YIELD_PER_GREAT_WORK_IN_CITY',
    GRANT_YIELD_PER_RESOURCES_IN_CITY: 'EFFECT_GRANT_YIELD_PER_RESOURCES_IN_CITY',
    GRANT_YIELD_PER_XP_EARNED: 'EFFECT_GRANT_YIELD_PER_XP_EARNED',
    MAP_REMOVE_CLIFFS_IN_DIRECTION: 'EFFECT_MAP_REMOVE_CLIFFS_IN_DIRECTION',
    METAPROGRESSION_COMPLETE_CHALLENGE: 'EFFECT_METAPROGRESSION_COMPLETE_CHALLENGE',
    METAPROGRESSION_SEND_EVENT: 'EFFECT_METAPROGRESSION_SEND_EVENT',
    MODIFY_CITY_YIELD_FOR_X_TURNS: 'EFFECT_MODIFY_CITY_YIELD_FOR_X_TURNS',
    MODIFY_PLAYER_TRADE_YIELD_CONVERSION: 'EFFECT_MODIFY_PLAYER_TRADE_YIELD_CONVERSION',
    MODIFY_PLAYER_YIELD_FOR_X_TURNS: 'EFFECT_MODIFY_PLAYER_YIELD_FOR_X_TURNS',
    PLAYER_ADD_GOLDEN_AGE_CHOICE: 'EFFECT_PLAYER_ADD_GOLDEN_AGE_CHOICE',
    PLAYER_ADD_YIELD_DEBT: 'EFFECT_PLAYER_ADD_YIELD_DEBT',
    PLAYER_ADD_YIELD_DEBT_DISCOVERY: 'EFFECT_PLAYER_ADD_YIELD_DEBT_DISCOVERY',
    PLAYER_ADD_YIELD_DEBT_NARRATIVE: 'EFFECT_PLAYER_ADD_YIELD_DEBT_NARRATIVE',
    PLAYER_ADJUST_AGE_PROGRESS: 'EFFECT_PLAYER_ADJUST_AGE_PROGRESS',
    PLAYER_ADJUST_ARTIFACT_EXTRACTION_SPEED: 'EFFECT_PLAYER_ADJUST_ARTIFACT_EXTRACTION_SPEED',
    PLAYER_ADJUST_CONSTRUCTIBLE_ADJACENCY: 'EFFECT_PLAYER_ADJUST_CONSTRUCTIBLE_ADJACENCY',
    PLAYER_ADJUST_CONSTRUCTIBLE_YIELD: 'EFFECT_PLAYER_ADJUST_CONSTRUCTIBLE_YIELD',
    PLAYER_ADJUST_CONSTRUCTIBLE_YIELD_BY_ATTRIBUTE: 'EFFECT_PLAYER_ADJUST_CONSTRUCTIBLE_YIELD_BY_ATTRIBUTE',
    PLAYER_ADJUST_FREE_CONSTRUCTIBLE_TIER: 'EFFECT_PLAYER_ADJUST_FREE_CONSTRUCTIBLE_TIER',
    PLAYER_ADJUST_GOLDEN_AGE_DURATION: 'EFFECT_PLAYER_ADJUST_GOLDEN_AGE_DURATION',
    PLAYER_ADJUST_GOLDEN_AGE_THRESHOLD: 'EFFECT_PLAYER_ADJUST_GOLDEN_AGE_THRESHOLD',
    PLAYER_ADJUST_LEGACY_PATH_GREAT_WORK_SCORING: 'EFFECT_PLAYER_ADJUST_LEGACY_PATH_GREAT_WORK_SCORING',
    PLAYER_ADJUST_LEGACY_PATH_HIGH_YIELD_DISTRICT_SCORING: 'EFFECT_PLAYER_ADJUST_LEGACY_PATH_HIGH_YIELD_DISTRICT_SCORING',
    PLAYER_ADJUST_LEGACY_PATH_NATURAL_WONDER_SCORING: 'EFFECT_PLAYER_ADJUST_LEGACY_PATH_NATURAL_WONDER_SCORING',
    PLAYER_ADJUST_LEGACY_PATH_RESOURCE_SCORING: 'EFFECT_PLAYER_ADJUST_LEGACY_PATH_RESOURCE_SCORING',
    PLAYER_ADJUST_LEGACY_PATH_SETTLEMENT_COUNT_SCORING: 'EFFECT_PLAYER_ADJUST_LEGACY_PATH_SETTLEMENT_COUNT_SCORING',
    PLAYER_ADJUST_LEGACY_PATH_SETTLEMENT_SCORING: 'EFFECT_PLAYER_ADJUST_LEGACY_PATH_SETTLEMENT_SCORING',
    PLAYER_ADJUST_LEGACY_PATH_WONDER_SCORING: 'EFFECT_PLAYER_ADJUST_LEGACY_PATH_WONDER_SCORING',
    PLAYER_ADJUST_PROGRESSION_TREE_EFFICIENCY: 'EFFECT_PLAYER_ADJUST_PROGRESSION_TREE_EFFICIENCY',
    PLAYER_ADJUST_PROGRESSION_TREE_MASTERY_EFFICIENCY: 'EFFECT_PLAYER_ADJUST_PROGRESSION_TREE_MASTERY_EFFICIENCY',
    PLAYER_ADJUST_PURCHASE_EFFICIENCY_PER_RESOURCE: 'EFFECT_PLAYER_ADJUST_PURCHASE_EFFICIENCY_PER_RESOURCE',
    PLAYER_ADJUST_SETTLEMENT_CAP: 'EFFECT_PLAYER_ADJUST_SETTLEMENT_CAP',
    PLAYER_ADJUST_TRADE_CAPACITY: 'EFFECT_PLAYER_ADJUST_TRADE_CAPACITY',
    PLAYER_ADJUST_UNIT_EXTRA_COPY: 'EFFECT_PLAYER_ADJUST_UNIT_EXTRA_COPY',
    PLAYER_ADJUST_UNIT_MAINTENANCE_EFFICIENCY: 'EFFECT_PLAYER_ADJUST_UNIT_MAINTENANCE_EFFICIENCY',
    PLAYER_ADJUST_UNIT_PRODUCTION_PER_GREAT_WORK: 'EFFECT_PLAYER_ADJUST_UNIT_PRODUCTION_PER_GREAT_WORK',
    PLAYER_ADJUST_UNIT_PRODUCTION_PER_RESOURCE: 'EFFECT_PLAYER_ADJUST_UNIT_PRODUCTION_PER_RESOURCE',
    PLAYER_ADJUST_UNIT_UPGRADE_DISCOUNT: 'EFFECT_PLAYER_ADJUST_UNIT_UPGRADE_DISCOUNT',
    PLAYER_ADJUST_YIELD: 'EFFECT_PLAYER_ADJUST_YIELD',
    PLAYER_ADJUST_YIELD_CONVERSION: 'EFFECT_PLAYER_ADJUST_YIELD_CONVERSION',
    PLAYER_ADJUST_YIELD_FOR_BECOMING_SUZERAIN: 'EFFECT_PLAYER_ADJUST_YIELD_FOR_BECOMING_SUZERAIN',
    PLAYER_ADJUST_YIELD_FOR_COMPLETING_NARRATIVE_EVENTS: 'EFFECT_PLAYER_ADJUST_YIELD_FOR_COMPLETING_NARRATIVE_EVENTS',
    PLAYER_ADJUST_YIELD_FOR_GOLDEN_AGE_START: 'EFFECT_PLAYER_ADJUST_YIELD_FOR_GOLDEN_AGE_START',
    PLAYER_ADJUST_YIELD_FROM_DISTATERS: 'EFFECT_PLAYER_ADJUST_YIELD_FROM_DISTATERS',
    PLAYER_ADJUST_YIELD_PER_ACTIVE_TRADITION: 'EFFECT_PLAYER_ADJUST_YIELD_PER_ACTIVE_TRADITION',
    PLAYER_ADJUST_YIELD_PER_ATTRIBUTE: 'EFFECT_PLAYER_ADJUST_YIELD_PER_ATTRIBUTE',
    PLAYER_ADJUST_YIELD_PER_ATTRIBUTE_AND_ALLIANCES: 'EFFECT_PLAYER_ADJUST_YIELD_PER_ATTRIBUTE_AND_ALLIANCES',
    PLAYER_ADJUST_YIELD_PER_ATTRIBUTE_LEVEL: 'EFFECT_PLAYER_ADJUST_YIELD_PER_ATTRIBUTE_LEVEL',
    PLAYER_ADJUST_YIELD_PER_COMPLETED_MASTERY: 'EFFECT_PLAYER_ADJUST_YIELD_PER_COMPLETED_MASTERY',
    PLAYER_ADJUST_YIELD_PER_CONQUERED_CITY: 'EFFECT_PLAYER_ADJUST_YIELD_PER_CONQUERED_CITY',
    PLAYER_ADJUST_YIELD_PER_CONQUERED_SETTLEMENT: 'EFFECT_PLAYER_ADJUST_YIELD_PER_CONQUERED_SETTLEMENT',
    PLAYER_ADJUST_YIELD_PER_NUM_CITIES: 'EFFECT_PLAYER_ADJUST_YIELD_PER_NUM_CITIES',
    PLAYER_ADJUST_YIELD_PER_NUM_CIVS_TRADING_WITH: 'EFFECT_PLAYER_ADJUST_YIELD_PER_NUM_CIVS_TRADING_WITH',
    PLAYER_ADJUST_YIELD_PER_NUM_TRADE_ROUTES: 'EFFECT_PLAYER_ADJUST_YIELD_PER_NUM_TRADE_ROUTES',
    PLAYER_ADJUST_YIELD_PER_PROPERTY_VALUE: 'EFFECT_PLAYER_ADJUST_YIELD_PER_PROPERTY_VALUE',
    PLAYER_ADJUST_YIELD_PER_RESOURCE: 'EFFECT_PLAYER_ADJUST_YIELD_PER_RESOURCE',
    PLAYER_ADJUST_YIELD_PER_RESOURCE_TYPE: 'EFFECT_PLAYER_ADJUST_YIELD_PER_RESOURCE_TYPE',
    PLAYER_ADJUST_YIELD_PER_SUZERAIN: 'EFFECT_PLAYER_ADJUST_YIELD_PER_SUZERAIN',
    PLAYER_ADJUST_YIELD_PER_UNIQUE_CIV_CONQUERED_CITY: 'EFFECT_PLAYER_ADJUST_YIELD_PER_UNIQUE_CIV_CONQUERED_CITY',
    PLAYER_ADJUST_YIELD_PER_UNIQUE_RESOURCE: 'EFFECT_PLAYER_ADJUST_YIELD_PER_UNIQUE_RESOURCE',
    PLAYER_ADJUST_YIELD_PER_UNIT_LEVEL: 'EFFECT_PLAYER_ADJUST_YIELD_PER_UNIT_LEVEL',
    PLAYER_ADJUST_YIELDS_PER_SETTLEMENT_OVER_CAP: 'EFFECT_PLAYER_ADJUST_YIELDS_PER_SETTLEMENT_OVER_CAP',
    PLAYER_ARMY_CAP: 'EFFECT_PLAYER_ARMY_CAP',
    PLAYER_ATTRIBUTE: 'EFFECT_PLAYER_ATTRIBUTE',
    PLAYER_ATTRIBUTE_UPGRADE: 'EFFECT_PLAYER_ATTRIBUTE_UPGRADE',
    PLAYER_BLOCK_GOVERNMENT_SELECTION: 'EFFECT_PLAYER_BLOCK_GOVERNMENT_SELECTION',
    PLAYER_DEDUCT_TREASURY: 'EFFECT_PLAYER_DEDUCT_TREASURY',
    PLAYER_DEDUCT_TREASURY_DISCOVERY: 'EFFECT_PLAYER_DEDUCT_TREASURY_DISCOVERY',
    PLAYER_DEDUCT_TREASURY_NARRATIVE: 'EFFECT_PLAYER_DEDUCT_TREASURY_NARRATIVE',
    PLAYER_DIPLOMACY_ACTION_BASIC_DIPLOMACY_MODS: 'EFFECT_PLAYER_DIPLOMACY_ACTION_BASIC_DIPLOMACY_MODS',
    PLAYER_DIPLOMACY_ACTION_DESCRIBE_PROGRESS_ENVOY_MOD: 'EFFECT_PLAYER_DIPLOMACY_ACTION_DESCRIBE_PROGRESS_ENVOY_MOD',
    PLAYER_DIPLOMACY_ACTION_GROUP_MOD: 'EFFECT_PLAYER_DIPLOMACY_ACTION_GROUP_MOD',
    PLAYER_DIPLOMACY_ACTION_PROGRESS_ENVOY_MOD: 'EFFECT_PLAYER_DIPLOMACY_ACTION_PROGRESS_ENVOY_MOD',
    PLAYER_DIPLOMACY_ACTION_PROGRESS_MOD: 'EFFECT_PLAYER_DIPLOMACY_ACTION_PROGRESS_MOD',
    PLAYER_DIPLOMACY_ADJUST_HAS_BOOSTED_SUPPORT_EFFECTS: 'EFFECT_PLAYER_DIPLOMACY_ADJUST_HAS_BOOSTED_SUPPORT_EFFECTS',
    PLAYER_DIPLOMACY_ADJUST_THIRD_PARTY_SUPPORT_MAX: 'EFFECT_PLAYER_DIPLOMACY_ADJUST_THIRD_PARTY_SUPPORT_MAX',
    PLAYER_DIPLOMACY_ALLOW_LEVY_FROM_ALL_CITY_STATES: 'EFFECT_PLAYER_DIPLOMACY_ALLOW_LEVY_FROM_ALL_CITY_STATES',
    PLAYER_DIPLOMACY_CHANGE_TRADE_ROUTE_CAPACITY: 'EFFECT_PLAYER_DIPLOMACY_CHANGE_TRADE_ROUTE_CAPACITY',
    PLAYER_DIPLOMACY_FAVOR_ACCRUE_MOD: 'EFFECT_PLAYER_DIPLOMACY_FAVOR_ACCRUE_MOD',
    PLAYER_DIPLOMACY_IGNORE_WAR_REQUIREMENTS: 'EFFECT_PLAYER_DIPLOMACY_IGNORE_WAR_REQUIREMENTS',
    PLAYER_DIPLOMACY_INCITE_RAID: 'EFFECT_PLAYER_DIPLOMACY_INCITE_RAID',
    PLAYER_DISABLE_BUILD_UNIT: 'EFFECT_PLAYER_DISABLE_BUILD_UNIT',
    PLAYER_FORCE_MILITARY_OPERATION: 'EFFECT_PLAYER_FORCE_MILITARY_OPERATION',
    PLAYER_GOVERNOR_CAP: 'EFFECT_PLAYER_GOVERNOR_CAP',
    PLAYER_GRANT_ATTRIBUTE_TREE_NODE: 'EFFECT_PLAYER_GRANT_ATTRIBUTE_TREE_NODE',
    PLAYER_GRANT_BONUS_STORIES: 'EFFECT_PLAYER_GRANT_BONUS_STORIES',
    PLAYER_GRANT_CONSTRUCTIBLE_UNLOCK: 'EFFECT_PLAYER_GRANT_CONSTRUCTIBLE_UNLOCK',
    PLAYER_GRANT_FAVORS_GRIEVANCES: 'EFFECT_PLAYER_GRANT_FAVORS_GRIEVANCES',
    PLAYER_GRANT_FREE_PURCHASE_CHARGE: 'EFFECT_PLAYER_GRANT_FREE_PURCHASE_CHARGE',
    PLAYER_GRANT_GIFT: 'EFFECT_PLAYER_GRANT_GIFT',
    PLAYER_GRANT_GOLDEN_AGE: 'EFFECT_PLAYER_GRANT_GOLDEN_AGE',
    PLAYER_GRANT_LEGACY_POINTS: 'EFFECT_PLAYER_GRANT_LEGACY_POINTS',
    PLAYER_GRANT_NARRATIVE_TAG: 'EFFECT_PLAYER_GRANT_NARRATIVE_TAG',
    PLAYER_GRANT_NAVAL_UNIT_WITH_OCEAN_ACCESS: 'EFFECT_PLAYER_GRANT_NAVAL_UNIT_WITH_OCEAN_ACCESS',
    PLAYER_GRANT_PROGRESSION: 'EFFECT_PLAYER_GRANT_PROGRESSION',
    PLAYER_GRANT_TRADITION_SLOTS: 'EFFECT_PLAYER_GRANT_TRADITION_SLOTS',
    PLAYER_GRANT_UNASSIGNED_WORKERS: 'EFFECT_PLAYER_GRANT_UNASSIGNED_WORKERS',
    PLAYER_GRANT_UNIT_AT_PLOT: 'EFFECT_PLAYER_GRANT_UNIT_AT_PLOT',
    PLAYER_GRANT_UNIT_ON_CAPTURE_CITY: 'EFFECT_PLAYER_GRANT_UNIT_ON_CAPTURE_CITY',
    PLAYER_GRANT_UNLOCK: 'EFFECT_PLAYER_GRANT_UNLOCK',
    PLAYER_GRANT_VICTORY_POINT: 'EFFECT_PLAYER_GRANT_VICTORY_POINT',
    PLAYER_GRANT_WONDER_APPEAL: 'EFFECT_PLAYER_GRANT_WONDER_APPEAL',
    PLAYER_GRANT_WONDER_PURCHASING: 'EFFECT_PLAYER_GRANT_WONDER_PURCHASING',
    PLAYER_GRANT_YIELD: 'EFFECT_PLAYER_GRANT_YIELD',
    PLAYER_GRANT_YIELD_ASSIGNED_WORKERS: 'EFFECT_PLAYER_GRANT_YIELD_ASSIGNED_WORKERS',
    PLAYER_GRANT_YIELD_DISCOVERY: 'EFFECT_PLAYER_GRANT_YIELD_DISCOVERY',
    PLAYER_GRANT_YIELD_NARRATIVE: 'EFFECT_PLAYER_GRANT_YIELD_NARRATIVE',
    PLAYER_LOWER_ALL_INDEPENDENT_RELATIONSHIPS: 'EFFECT_PLAYER_LOWER_ALL_INDEPENDENT_RELATIONSHIPS',
    PLAYER_OPEN_ARCHAEOLOGY: 'EFFECT_PLAYER_OPEN_ARCHAEOLOGY',
    PLAYER_PROPERTY: 'EFFECT_PLAYER_PROPERTY',
    PLAYER_REPLACE_CONSTRUCTIBLE: 'EFFECT_PLAYER_REPLACE_CONSTRUCTIBLE',
    PLAYER_REVEAL_CULTURE_TREE: 'EFFECT_PLAYER_REVEAL_CULTURE_TREE',
    PLAYER_SAME_GOVERNMENT_GRANT_FAVORS_GRIEVANCES: 'EFFECT_PLAYER_SAME_GOVERNMENT_GRANT_FAVORS_GRIEVANCES',
    PLAYER_SET_GOVERNMENT: 'EFFECT_PLAYER_SET_GOVERNMENT',
    PLAYER_SPAWN_TRIBE_ARMY: 'EFFECT_PLAYER_SPAWN_TRIBE_ARMY',
    PLAYER_SPAWN_TRIBE_NEAR: 'EFFECT_PLAYER_SPAWN_TRIBE_NEAR',
    PLAYER_UNLOCK_PANTHEON: 'EFFECT_PLAYER_UNLOCK_PANTHEON',
    PLAYER_UNLOCK_TRADITION: 'EFFECT_PLAYER_UNLOCK_TRADITION',
    PLAYER_WAR_DIPLOMATIC_RELATIONSHIP_CHANGE: 'EFFECT_PLAYER_WAR_DIPLOMATIC_RELATIONSHIP_CHANGE',
    PLOT_ADJUST_WORKER_MAINTENANCE: 'EFFECT_PLOT_ADJUST_WORKER_MAINTENANCE',
    PLOT_ADJUST_WORKER_YIELD: 'EFFECT_PLOT_ADJUST_WORKER_YIELD',
    PLOT_ADJUST_YIELD: 'EFFECT_PLOT_ADJUST_YIELD',
    PLOT_ADJUST_YIELD_RANDOM: 'EFFECT_PLOT_ADJUST_YIELD_RANDOM',
    PLOT_DESTROY_DISCOVERY: 'EFFECT_PLOT_DESTROY_DISCOVERY',
    PLOT_PLACE_RESOURCE: 'EFFECT_PLOT_PLACE_RESOURCE',
    PLOTS_IN_RANGE_ADJUST_YIELD: 'EFFECT_PLOTS_IN_RANGE_ADJUST_YIELD',
    PROMOTE_TOWN_TO_CITY: 'EFFECT_PROMOTE_TOWN_TO_CITY',
    RESTORE_UNIT_MOVEMENT: 'EFFECT_RESTORE_UNIT_MOVEMENT',
    REVEAL_UNIT_PLOT_FOR_PLAYER: 'EFFECT_REVEAL_UNIT_PLOT_FOR_PLAYER',
    START_LAND_CLAIM: 'EFFECT_START_LAND_CLAIM',
    TIMEOUT_UNIT_OFF_MAP: 'EFFECT_TIMEOUT_UNIT_OFF_MAP',
    TREAT_CAPITAL_AS_HOLY_CITY: 'EFFECT_TREAT_CAPITAL_AS_HOLY_CITY',
    TRIGGER_GAME_MECHANIC: 'EFFECT_TRIGGER_GAME_MECHANIC',
    UNIT_ADJUST_ABILITY: 'EFFECT_UNIT_ADJUST_ABILITY',
    UNIT_ADJUST_COMBAT_STRENGTH_PER_GAME_DIFFICULTY: 'EFFECT_UNIT_ADJUST_COMBAT_STRENGTH_PER_GAME_DIFFICULTY',
    UNIT_ADJUST_COMBAT_STRENGTH_PER_RESOURCE: 'EFFECT_UNIT_ADJUST_COMBAT_STRENGTH_PER_RESOURCE',
    UNIT_ADJUST_COMMAND_AWARD: 'EFFECT_UNIT_ADJUST_COMMAND_AWARD',
    UNIT_ADJUST_DAMAGE: 'EFFECT_UNIT_ADJUST_DAMAGE',
    UNIT_ADJUST_EMBARKATION_TYPE: 'EFFECT_UNIT_ADJUST_EMBARKATION_TYPE',
    UNIT_ADJUST_FLANK_DEFENSE_MODIFIER: 'EFFECT_UNIT_ADJUST_FLANK_DEFENSE_MODIFIER',
    UNIT_ADJUST_FLANKING_ATTACK_MODIFIER: 'EFFECT_UNIT_ADJUST_FLANKING_ATTACK_MODIFIER',
    UNIT_ADJUST_FORTIFICATION_CONSTRUCTION_TIME: 'EFFECT_UNIT_ADJUST_FORTIFICATION_CONSTRUCTION_TIME',
    UNIT_ADJUST_HEAL_PER_RESOURCE: 'EFFECT_UNIT_ADJUST_HEAL_PER_RESOURCE',
    UNIT_ADJUST_HEAL_PER_TURN: 'EFFECT_UNIT_ADJUST_HEAL_PER_TURN',
    UNIT_ADJUST_IGNORE_MOVEMENT_OBSTACLE: 'EFFECT_UNIT_ADJUST_IGNORE_MOVEMENT_OBSTACLE',
    UNIT_ADJUST_MOVE_AFTER_ATTACK: 'EFFECT_UNIT_ADJUST_MOVE_AFTER_ATTACK',
    UNIT_ADJUST_MOVEMENT: 'EFFECT_UNIT_ADJUST_MOVEMENT',
    UNIT_ADJUST_OPERATION_AWARD: 'EFFECT_UNIT_ADJUST_OPERATION_AWARD',
    UNIT_ADJUST_PLAYER_YIELD: 'EFFECT_UNIT_ADJUST_PLAYER_YIELD',
    UNIT_GRANT_FREE_BUILDING_IGNORE: 'EFFECT_UNIT_GRANT_FREE_BUILDING_IGNORE',
    UNIT_GRANT_UNASSIGNED_WORKERS: 'EFFECT_UNIT_GRANT_UNASSIGNED_WORKERS',
    UNIT_PROPERTY: 'EFFECT_UNIT_PROPERTY',
    UNIT_SET_ATTACK_AFTER_MOVE: 'EFFECT_UNIT_SET_ATTACK_AFTER_MOVE',
    UNIT_SET_TRADE_ROUTE_PROTECTED: 'EFFECT_UNIT_SET_TRADE_ROUTE_PROTECTED',
    UNITS_EXTRA_RANDOM_EVENT_DAMAGE: 'EFFECT_UNITS_EXTRA_RANDOM_EVENT_DAMAGE',
    UNITS_IMMUNE_TO_RANDOM_EVENTS: 'EFFECT_UNITS_IMMUNE_TO_RANDOM_EVENTS',
};
