// civ7-modding-tools@1.3.0/dist/constants/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.YIELD = exports.UNIT_MOVEMENT_CLASS = exports.UNIT_CULTURE = exports.UNIT_CLASS = exports.UNIT = exports.TRAIT = exports.TERRAIN = exports.TAG_TRAIT = exports.SYSTEM = exports.RESOURCE = exports.REQUIREMENT_SET = exports.REQUIREMENT = exports.PLUNDER = exports.LANGUAGE = exports.KIND = exports.ICON = exports.FORMATION_CLASS = exports.FERTILITY = exports.FEATURE_CLASS = exports.FEATURE = exports.EFFECT = exports.DOMAIN = exports.DISTRICT = exports.CORE_CLASS = exports.CONSTRUCTIBLE_TYPE_TAG = exports.CONSTRUCTIBLE_CLASS = exports.COLLECTION = exports.CIVILIZATION_DOMAIN = exports.BUILDING_CULTURES = exports.BIOME = exports.AGE = exports.ADVISORY = exports.ACTION_GROUP_BUNDLE = exports.ACTION_GROUP_ACTION = exports.ACTION_GROUP = void 0;
var ACTION_GROUP_1 = require("./ACTION_GROUP");
Object.defineProperty(exports, "ACTION_GROUP", { enumerable: true, get: function () { return ACTION_GROUP_1.ACTION_GROUP; } });
var ACTION_GROUP_ACTION_1 = require("./ACTION_GROUP_ACTION");
Object.defineProperty(exports, "ACTION_GROUP_ACTION", { enumerable: true, get: function () { return ACTION_GROUP_ACTION_1.ACTION_GROUP_ACTION; } });
var ACTION_GROUP_BUNDLE_1 = require("./ACTION_GROUP_BUNDLE");
Object.defineProperty(exports, "ACTION_GROUP_BUNDLE", { enumerable: true, get: function () { return ACTION_GROUP_BUNDLE_1.ACTION_GROUP_BUNDLE; } });
var ADVISORY_1 = require("./ADVISORY");
Object.defineProperty(exports, "ADVISORY", { enumerable: true, get: function () { return ADVISORY_1.ADVISORY; } });
var AGE_1 = require("./AGE");
Object.defineProperty(exports, "AGE", { enumerable: true, get: function () { return AGE_1.AGE; } });
var BIOME_1 = require("./BIOME");
Object.defineProperty(exports, "BIOME", { enumerable: true, get: function () { return BIOME_1.BIOME; } });
var BUILDING_CULTURES_1 = require("./BUILDING_CULTURES");
Object.defineProperty(exports, "BUILDING_CULTURES", { enumerable: true, get: function () { return BUILDING_CULTURES_1.BUILDING_CULTURES; } });
var CIVILIZATION_DOMAIN_1 = require("./CIVILIZATION_DOMAIN");
Object.defineProperty(exports, "CIVILIZATION_DOMAIN", { enumerable: true, get: function () { return CIVILIZATION_DOMAIN_1.CIVILIZATION_DOMAIN; } });
var COLLECTION_1 = require("./COLLECTION");
Object.defineProperty(exports, "COLLECTION", { enumerable: true, get: function () { return COLLECTION_1.COLLECTION; } });
var CONSTRUCTIBLE_CLASS_1 = require("./CONSTRUCTIBLE_CLASS");
Object.defineProperty(exports, "CONSTRUCTIBLE_CLASS", { enumerable: true, get: function () { return CONSTRUCTIBLE_CLASS_1.CONSTRUCTIBLE_CLASS; } });
var CONSTRUCTIBLE_TYPE_TAG_1 = require("./CONSTRUCTIBLE_TYPE_TAG");
Object.defineProperty(exports, "CONSTRUCTIBLE_TYPE_TAG", { enumerable: true, get: function () { return CONSTRUCTIBLE_TYPE_TAG_1.CONSTRUCTIBLE_TYPE_TAG; } });
var CORE_CLASS_1 = require("./CORE_CLASS");
Object.defineProperty(exports, "CORE_CLASS", { enumerable: true, get: function () { return CORE_CLASS_1.CORE_CLASS; } });
var DISTRICT_1 = require("./DISTRICT");
Object.defineProperty(exports, "DISTRICT", { enumerable: true, get: function () { return DISTRICT_1.DISTRICT; } });
var DOMAIN_1 = require("./DOMAIN");
Object.defineProperty(exports, "DOMAIN", { enumerable: true, get: function () { return DOMAIN_1.DOMAIN; } });
var EFFECT_1 = require("./EFFECT");
Object.defineProperty(exports, "EFFECT", { enumerable: true, get: function () { return EFFECT_1.EFFECT; } });
var FEATURE_1 = require("./FEATURE");
Object.defineProperty(exports, "FEATURE", { enumerable: true, get: function () { return FEATURE_1.FEATURE; } });
var FEATURE_CLASS_1 = require("./FEATURE_CLASS");
Object.defineProperty(exports, "FEATURE_CLASS", { enumerable: true, get: function () { return FEATURE_CLASS_1.FEATURE_CLASS; } });
var FERTILITY_1 = require("./FERTILITY");
Object.defineProperty(exports, "FERTILITY", { enumerable: true, get: function () { return FERTILITY_1.FERTILITY; } });
var FORMATION_CLASS_1 = require("./FORMATION_CLASS");
Object.defineProperty(exports, "FORMATION_CLASS", { enumerable: true, get: function () { return FORMATION_CLASS_1.FORMATION_CLASS; } });
var ICON_1 = require("./ICON");
Object.defineProperty(exports, "ICON", { enumerable: true, get: function () { return ICON_1.ICON; } });
var KIND_1 = require("./KIND");
Object.defineProperty(exports, "KIND", { enumerable: true, get: function () { return KIND_1.KIND; } });
var LANGUAGE_1 = require("./LANGUAGE");
Object.defineProperty(exports, "LANGUAGE", { enumerable: true, get: function () { return LANGUAGE_1.LANGUAGE; } });
var PLUNDER_1 = require("./PLUNDER");
Object.defineProperty(exports, "PLUNDER", { enumerable: true, get: function () { return PLUNDER_1.PLUNDER; } });
var REQUIREMENT_1 = require("./REQUIREMENT");
Object.defineProperty(exports, "REQUIREMENT", { enumerable: true, get: function () { return REQUIREMENT_1.REQUIREMENT; } });
var REQUIREMENT_SET_1 = require("./REQUIREMENT_SET");
Object.defineProperty(exports, "REQUIREMENT_SET", { enumerable: true, get: function () { return REQUIREMENT_SET_1.REQUIREMENT_SET; } });
var RESOURCE_1 = require("./RESOURCE");
Object.defineProperty(exports, "RESOURCE", { enumerable: true, get: function () { return RESOURCE_1.RESOURCE; } });
var SYSTEM_1 = require("./SYSTEM");
Object.defineProperty(exports, "SYSTEM", { enumerable: true, get: function () { return SYSTEM_1.SYSTEM; } });
var TAG_TRAIT_1 = require("./TAG_TRAIT");
Object.defineProperty(exports, "TAG_TRAIT", { enumerable: true, get: function () { return TAG_TRAIT_1.TAG_TRAIT; } });
var TERRAIN_1 = require("./TERRAIN");
Object.defineProperty(exports, "TERRAIN", { enumerable: true, get: function () { return TERRAIN_1.TERRAIN; } });
var TRAIT_1 = require("./TRAIT");
Object.defineProperty(exports, "TRAIT", { enumerable: true, get: function () { return TRAIT_1.TRAIT; } });
var UNIT_1 = require("./UNIT");
Object.defineProperty(exports, "UNIT", { enumerable: true, get: function () { return UNIT_1.UNIT; } });
var UNIT_CLASS_1 = require("./UNIT_CLASS");
Object.defineProperty(exports, "UNIT_CLASS", { enumerable: true, get: function () { return UNIT_CLASS_1.UNIT_CLASS; } });
var UNIT_CULTURE_1 = require("./UNIT_CULTURE");
Object.defineProperty(exports, "UNIT_CULTURE", { enumerable: true, get: function () { return UNIT_CULTURE_1.UNIT_CULTURE; } });
var UNIT_MOVEMENT_CLASS_1 = require("./UNIT_MOVEMENT_CLASS");
Object.defineProperty(exports, "UNIT_MOVEMENT_CLASS", { enumerable: true, get: function () { return UNIT_MOVEMENT_CLASS_1.UNIT_MOVEMENT_CLASS; } });
var YIELD_1 = require("./YIELD");
Object.defineProperty(exports, "YIELD", { enumerable: true, get: function () { return YIELD_1.YIELD; } });
