// civ7-modding-tools@1.3.0/dist/nodes/UnitNode.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UnitNode = void 0;
const constants_1 = require("../constants");
const BaseNode_1 = require("./BaseNode");
class UnitNode extends BaseNode_1.BaseNode {
    constructor(payload = {}) {
        super();
        this.unitType = 'UNIT_TYPE';
        this.baseMoves = 1;
        this.baseSightRange = 1;
        this.name = 'UNIT_NAME';
        this.coreClass = constants_1.CORE_CLASS.MILITARY;
        this.domain = constants_1.DOMAIN.LAND;
        this.formationClass = constants_1.FORMATION_CLASS.LAND_COMBAT;
        this.unitMovementClass = constants_1.UNIT_MOVEMENT_CLASS.FOOT;
        this.airSlots = null;
        this.allowBarbarians = null;
        this.allowTeleportToOtherPlayerCapitals = null;
        this.antiAirCombat = null;
        this.buildCharges = null;
        this.canBeDamaged = null;
        this.canCapture = null;
        this.canEarnExperience = null;
        this.canPurchase = null;
        this.canRetreatWhenCaptured = null;
        this.canTargetAir = null;
        this.canTrain = null;
        this.canTriggerDiscovery = null;
        this.costProgressionModel = null;
        this.costProgressionParam1 = null;
        this.description = null;
        this.enabledByReligion = null;
        this.evangelizeBelief = null;
        this.extractsArtifacts = null;
        this.foundCity = null;
        this.foundReligion = null;
        this.ignoreMoves = null;
        this.initialLevel = null;
        this.launchInquisition = null;
        this.maintenance = null;
        this.makeTradeRoute = null;
        this.mustPurchase = null;
        this.numRandomChoices = null;
        this.prereqPopulation = null;
        this.promotionClass = null;
        this.pseudoYieldType = null;
        this.purchaseYield = null;
        this.religionEvictPercent = null;
        this.religiousHealCharges = null;
        this.religiousStrength = null;
        this.requiresInquisition = null;
        this.spreadCharges = null;
        this.spy = null;
        this.stackable = null;
        this.strategicResource = null;
        this.teamVisibility = null;
        this.teleport = null;
        this.tier = null;
        this.trackReligion = null;
        this.traitType = null;
        this.victoryType = null;
        this.victoryUnit = null;
        this.wmdCapable = null;
        this.zoneOfControl = null;
        this.fill(payload);
    }
}
exports.UnitNode = UnitNode;
